Index: src/bcm/esw/vlan.c
===================================================================
RCS file: /wd/cvsroot/sdk-xgs-robo-6.3.0/src/bcm/esw/vlan.c,v
retrieving revision 1.1.1.1
diff -u -p -I '\$Id:' -r1.1.1.1 vlan.c
--- src/bcm/esw/vlan.c	29 Apr 2013 09:42:32 -0000	1.1.1.1
+++ src/bcm/esw/vlan.c	18 Sep 2013 06:54:01 -0000
@@ -5620,6 +5620,7 @@ bcm_esw_vlan_translate_egress_action_add
     bcm_module_t        modid;
     bcm_trunk_t         tgid = -1;
     int                 id = -1; 
+    int orig_port_class = port_class;
 
     CHECK_INIT(unit);
     VLAN_CHK_ID(unit, outer_vlan);
@@ -5639,7 +5640,8 @@ bcm_esw_vlan_translate_egress_action_add
         }
         if ((-1 != tgid || -1 != id) && !BCM_GPORT_IS_TUNNEL(port_class) &&
             !BCM_GPORT_IS_MIM_PORT(port_class) && 
-            !BCM_GPORT_IS_MPLS_PORT(port_class)) {
+            !BCM_GPORT_IS_MPLS_PORT(port_class) && 
+            !BCM_GPORT_IS_VLAN_PORT(orig_port_class)) {
             return BCM_E_PORT;
         }
     } else {
@@ -5650,6 +5652,9 @@ bcm_esw_vlan_translate_egress_action_add
 
 #ifdef BCM_TRX_SUPPORT
     if (SOC_IS_TRX(unit) && soc_feature(unit, soc_feature_vlan_action)) {
+        if(BCM_GPORT_IS_VLAN_PORT(orig_port_class)) {
+            port_class = orig_port_class;
+        }
         return _bcm_trx_vlan_translate_egress_action_add(unit, port_class, 
                                              outer_vlan, inner_vlan, 
                                              action);
@@ -5680,6 +5685,7 @@ bcm_esw_vlan_translate_egress_action_get
     bcm_module_t        modid;
     bcm_trunk_t         tgid = -1;
     int                 id = -1; 
+    int orig_port_class = port_class;
 
     CHECK_INIT(unit);
     VLAN_CHK_ID(unit, outer_vlan);
@@ -5698,8 +5704,9 @@ bcm_esw_vlan_translate_egress_action_get
                                        &tgid, &id));
         }
         if ((-1 != tgid || -1 != id) && !BCM_GPORT_IS_TUNNEL(port_class) &&
-            !BCM_GPORT_IS_MIM_PORT(port_class) &&
-            !BCM_GPORT_IS_MPLS_PORT(port_class)) {
+            !BCM_GPORT_IS_MIM_PORT(port_class) && 
+            !BCM_GPORT_IS_MPLS_PORT(port_class) && 
+            !BCM_GPORT_IS_VLAN_PORT(orig_port_class)) {
             return BCM_E_PORT;
         }
     } else {
@@ -5710,6 +5717,9 @@ bcm_esw_vlan_translate_egress_action_get
 
 #ifdef BCM_TRX_SUPPORT
     if (SOC_IS_TRX(unit) && soc_feature(unit, soc_feature_vlan_action)) {
+        if(BCM_GPORT_IS_VLAN_PORT(orig_port_class)) {
+            port_class = orig_port_class;
+        }
         return _bcm_trx_vlan_translate_egress_action_get(unit, port_class, 
                                                         outer_vlan, inner_vlan, 
                                                         action);
@@ -5738,6 +5748,7 @@ bcm_esw_vlan_translate_egress_action_del
     bcm_module_t        modid;
     bcm_trunk_t         tgid = -1;
     int                 id = -1; 
+    int orig_port_class = port_class;
 
     CHECK_INIT(unit);
     VLAN_CHK_ID(unit, outer_vlan);
@@ -5752,8 +5763,9 @@ bcm_esw_vlan_translate_egress_action_del
                                        &tgid, &id));
         }
         if ((-1 != tgid || -1 != id) && !BCM_GPORT_IS_TUNNEL(port_class) &&
-            !BCM_GPORT_IS_MIM_PORT(port_class) &&
-            !BCM_GPORT_IS_MPLS_PORT(port_class)) {
+            !BCM_GPORT_IS_MIM_PORT(port_class) && 
+            !BCM_GPORT_IS_MPLS_PORT(port_class) && 
+            !BCM_GPORT_IS_VLAN_PORT(orig_port_class)) {
             return BCM_E_PORT;
         }
     } else {
@@ -5764,6 +5776,9 @@ bcm_esw_vlan_translate_egress_action_del
 
 #ifdef BCM_TRX_SUPPORT
     if (SOC_IS_TRX(unit) && soc_feature(unit, soc_feature_vlan_action)) {
+        if(BCM_GPORT_IS_VLAN_PORT(orig_port_class)) {
+            port_class = orig_port_class;
+        }
         return _bcm_trx_vlan_translate_egress_action_delete(unit, port_class, 
                                                            outer_vlan, 
                                                            inner_vlan);
Index: src/bcm/esw/firebolt/vlan.c
===================================================================
RCS file: /wd/cvsroot/sdk-xgs-robo-6.3.0/src/bcm/esw/firebolt/vlan.c,v
retrieving revision 1.1.1.1
diff -u -p -I '\$Id:' -r1.1.1.1 vlan.c
--- src/bcm/esw/firebolt/vlan.c	29 Apr 2013 09:42:32 -0000	1.1.1.1
+++ src/bcm/esw/firebolt/vlan.c	18 Sep 2013 06:50:23 -0000
@@ -4525,6 +4525,18 @@ _bcm_xgs3_vlan_control_vlan_set(int unit
                                 _BCM_MULTICAST_ID_GET
                                 (control->unknown_unicast_group));
         }
+        
+        if (_BCM_MULTICAST_IS_VLAN(control->broadcast_group) || 
+            _BCM_MULTICAST_IS_VLAN(control->unknown_unicast_group) ||
+            _BCM_MULTICAST_IS_VLAN(control->unknown_unicast_group)) {
+            if (SOC_MEM_FIELD_VALID(unit, VLAN_TABm, VIRTUAL_PORT_ENf)) {
+                soc_mem_field32_set(unit, VLAN_TABm, &vt, VIRTUAL_PORT_ENf, 1);
+            }
+        } else {
+            if (SOC_MEM_FIELD_VALID(unit, VLAN_TABm, VIRTUAL_PORT_ENf)) {
+                soc_mem_field32_set(unit, VLAN_TABm, &vt, VIRTUAL_PORT_ENf, 0);
+            }
+        }
    }
 #endif /* (BCM_TRIUMPH2_SUPPORT && INCLUDE_L3 */
 
