Index: src/soc/dpp/ARAD/arad_pmf_low_level.c
===================================================================
--- src/soc/dpp/ARAD/arad_pmf_low_level.c	(revision 2075)
+++ src/soc/dpp/ARAD/arad_pmf_low_level.c	(revision 2076)
@@ -559,12 +559,13 @@ CONST STATIC
     {2, 1,11, SOC_PPC_FP_QUAL_HDR_ITMH_DEST_FWD,       16, 4, SOC_SAND_BIT(ARAD_PMF_KEY_D)}, /* Forwarding Destination - for System-Port */
     {2, 1,14, SOC_PPC_FP_QUAL_HDR_ITMH_EXT,            20, 4, SOC_SAND_BIT(ARAD_PMF_KEY_D)}, /* Destination Extension */
     {2, 1,15, SOC_PPC_FP_QUAL_IRPP_PARSER_LEAF_CONTEXT,     1, 0, SOC_SAND_BIT(ARAD_PMF_KEY_D)}, /* Parser-Leaf-Context bit 0 */
 };
 
 /* Jericho parsing of Jericho-ITMH */
+#if (__GNUC__ < 6) /* PTin removed: This is not used */
 CONST STATIC
     ARAD_PMF_KEY_CONSTR
         Jericho_pmf_key_construction_info_tm[] = {
             /* Regular TM Program */
             /* Is-Not-TM, Is-MSB, CE-id   Qualifier type,   Nof-bits, Qual-LSB,   Key-Bitmap        */
      /* KeyA: 160b after ITMH for TM and FP */
@@ -617,12 +618,13 @@ CONST STATIC
     {2, 1, 6, SOC_PPC_FP_QUAL_HDR_ITMH_DEST_FWD,       17, 3, SOC_SAND_BIT(ARAD_PMF_KEY_D)}, /* Forwarding Destination */
     {2, 1, 7, SOC_PPC_FP_QUAL_IRPP_PMF_PGM_KEY_GEN_VAR, 3,24, SOC_SAND_BIT(ARAD_PMF_KEY_D)}, /* System-Port prefix */
     {2, 1,11, SOC_PPC_FP_QUAL_HDR_ITMH_DEST_FWD,       16, 4, SOC_SAND_BIT(ARAD_PMF_KEY_D)}, /* Forwarding Destination - for System-Port */
     {2, 1,14, SOC_PPC_FP_QUAL_HDR_ITMH_EXT,            20, 4, SOC_SAND_BIT(ARAD_PMF_KEY_D)}, /* Destination Extension */
     {2, 1,15, SOC_PPC_FP_QUAL_IRPP_PARSER_LEAF_CONTEXT,     1, 0, SOC_SAND_BIT(ARAD_PMF_KEY_D)}, /* Parser-Leaf-Context bit 0 */
 };
+#endif
 
 CONST STATIC
     ARAD_PMF_KEY_CONSTR
         Arad_pmf_key_construction_info_mh[] = {
             /* Regular TM Program */
 /* Is-Not-TM, Is-MSB,CE-id,Qualifier type,   Nof-bits, Qual-LSB,   Key-Bitmap        */
@@ -866,12 +868,13 @@ CONST STATIC
   {2,19, {SOC_SAND_MAGIC_NUM_VAL,        5, SOC_PPC_FP_ACTION_TYPE_USER_HEADER_2, TRUE},             ARAD_PMF_KEY_B, ARAD_PMF_KEY_LSB_16}, /* take all 19 bit dest from itmh without encoding */
   {2,20, {SOC_SAND_MAGIC_NUM_VAL,        0, SOC_PPC_FP_ACTION_TYPE_USER_HEADER_2, FALSE},             ARAD_PMF_KEY_D, ARAD_PMF_KEY_LSB_80}, /* In case of MC_FLOW_ID or IS, override the User-Header-2 with the Destination Extension */
   {2,21, {SOC_SAND_MAGIC_NUM_VAL,        2, SOC_PPC_FP_ACTION_TYPE_USER_HEADER_2, FALSE},           ARAD_PMF_KEY_C, ARAD_PMF_KEY_LSB_0}, /* In case of Out-LIF ITMH Dest, override the User-header with Out-LIF */
 };
 
 /* Jericho parsing of Jericho-ITMH */
+#if (__GNUC__ < 6) /* PTin removed: This is not used */
 CONST STATIC
     ARAD_PMF_FES_CONSTR
         Jericho_pmf_fes_construction_info_tm[] = {
             /* Regular TM Program */
   /*  FES-ID,                       Shift  Action type,       Is-Always-Valid                      Key-source      Key LSB        */
   {2, 2, {SOC_SAND_MAGIC_NUM_VAL,       16, SOC_PPC_FP_ACTION_TYPE_DP, TRUE},                         ARAD_PMF_KEY_B, ARAD_PMF_KEY_LSB_32},
@@ -898,12 +901,13 @@ CONST STATIC
   {2,19, {SOC_SAND_MAGIC_NUM_VAL,        7, SOC_PPC_FP_ACTION_TYPE_SYSTEM_HEADER_PROFILE_ID, TRUE}, ARAD_PMF_KEY_C, ARAD_PMF_KEY_LSB_96}, /* header profile is coded like the destination prefix. take 3 msb (19:17) */
   {2,20, {SOC_SAND_MAGIC_NUM_VAL,        4, SOC_PPC_FP_ACTION_TYPE_SYSTEM_HEADER_PROFILE_ID, FALSE}, ARAD_PMF_KEY_B, ARAD_PMF_KEY_LSB_48}, /* header profile is coded like the destination prefix. take 3 msb (19:17) */
   {2,21, {SOC_SAND_MAGIC_NUM_VAL,        5, SOC_PPC_FP_ACTION_TYPE_USER_HEADER_2, TRUE},             ARAD_PMF_KEY_B, /* ARAD_PMF_KEY_LSB_16 */ ARAD_PMF_KEY_LSB_0}, /* take all 19 bit dest from itmh without encoding */
   {2,22, {SOC_SAND_MAGIC_NUM_VAL,        0, SOC_PPC_FP_ACTION_TYPE_USER_HEADER_2, FALSE},             ARAD_PMF_KEY_D, ARAD_PMF_KEY_LSB_80}, /* In case of MC_FLOW_ID or IS, override the User-Header-2 with the Destination Extension */
   {2,23, {SOC_SAND_MAGIC_NUM_VAL,        2, SOC_PPC_FP_ACTION_TYPE_USER_HEADER_2, FALSE},           ARAD_PMF_KEY_C, ARAD_PMF_KEY_LSB_0}, /* In case of Out-LIF ITMH Dest, override the User-header with Out-LIF */
 };
+#endif
 
 
 
 
 CONST STATIC
     ARAD_PMF_FES_CONSTR
