diff -rupN linux-denx-3.18.25.ori/arch/alpha/oprofile/Makefile linux-denx-3.18.25.mod/arch/alpha/oprofile/Makefile
--- linux-denx-3.18.25.ori/arch/alpha/oprofile/Makefile	2016-11-17 13:23:08.931075000 +0000
+++ linux-denx-3.18.25.mod/arch/alpha/oprofile/Makefile	2017-07-11 16:38:05.764712978 +0100
@@ -1,4 +1,4 @@
-ccflags-y := -Werror -Wno-sign-compare
+ccflags-y := -Wall -Wno-sign-compare
 
 obj-$(CONFIG_OPROFILE) += oprofile.o
 
diff -rupN linux-denx-3.18.25.ori/arch/powerpc/boot/crt0.S linux-denx-3.18.25.mod/arch/powerpc/boot/crt0.S
--- linux-denx-3.18.25.ori/arch/powerpc/boot/crt0.S	2016-11-17 13:38:47.915045000 +0000
+++ linux-denx-3.18.25.mod/arch/powerpc/boot/crt0.S	2017-07-11 16:38:05.764712978 +0100
@@ -24,20 +24,6 @@ RELACOUNT = 0x6ffffff9
 _zimage_start_opd:
 	.long	0x500000, 0, 0, 0
 
-#ifdef __powerpc64__
-.balign 8
-p_start:	.llong	_start
-p_etext:	.llong	_etext
-p_bss_start:	.llong	__bss_start
-p_end:		.llong	_end
-
-p_toc:		.llong	__toc_start + 0x8000 - p_base
-p_dyn:		.llong	__dynamic_start - p_base
-p_rela:		.llong	__rela_dyn_start - p_base
-p_prom:		.llong	0
-	.weak	_platform_stack_top
-p_pstack:	.llong	_platform_stack_top
-#else
 p_start:	.long	_start
 p_etext:	.long	_etext
 p_bss_start:	.long	__bss_start
@@ -45,7 +31,6 @@ p_end:		.long	_end
 
 	.weak	_platform_stack_top
 p_pstack:	.long	_platform_stack_top
-#endif
 
 	.weak	_zimage_start
 	.globl	_zimage_start
@@ -56,88 +41,6 @@ _zimage_start_lib:
 	   and the address where we're running. */
 	bl	.+4
 p_base:	mflr	r10		/* r10 now points to runtime addr of p_base */
-#ifndef __powerpc64__
-	/* grab the link address of the dynamic section in r11 */
-	addis	r11,r10,(_GLOBAL_OFFSET_TABLE_-p_base)@ha
-	lwz	r11,(_GLOBAL_OFFSET_TABLE_-p_base)@l(r11)
-	cmpwi	r11,0
-	beq	3f		/* if not linked -pie */
-	/* get the runtime address of the dynamic section in r12 */
-	.weak	__dynamic_start
-	addis	r12,r10,(__dynamic_start-p_base)@ha
-	addi	r12,r12,(__dynamic_start-p_base)@l
-	subf	r11,r11,r12	/* runtime - linktime offset */
-
-	/* The dynamic section contains a series of tagged entries.
-	 * We need the RELA and RELACOUNT entries. */
-	li	r9,0
-	li	r0,0
-9:	lwz	r8,0(r12)	/* get tag */
-	cmpwi	r8,0
-	beq	10f		/* end of list */
-	cmpwi	r8,RELA
-	bne	11f
-	lwz	r9,4(r12)	/* get RELA pointer in r9 */
-	b	12f
-11:	addis	r8,r8,(-RELACOUNT)@ha
-	cmpwi	r8,RELACOUNT@l
-	bne	12f
-	lwz	r0,4(r12)	/* get RELACOUNT value in r0 */
-12:	addi	r12,r12,8
-	b	9b
-
-	/* The relocation section contains a list of relocations.
-	 * We now do the R_PPC_RELATIVE ones, which point to words
-	 * which need to be initialized with addend + offset.
-	 * The R_PPC_RELATIVE ones come first and there are RELACOUNT
-	 * of them. */
-10:	/* skip relocation if we don't have both */
-	cmpwi	r0,0
-	beq	3f
-	cmpwi	r9,0
-	beq	3f
-
-	add	r9,r9,r11	/* Relocate RELA pointer */
-	mtctr	r0
-2:	lbz	r0,4+3(r9)	/* ELF32_R_INFO(reloc->r_info) */
-	cmpwi	r0,22		/* R_PPC_RELATIVE */
-	bne	3f
-	lwz	r12,0(r9)	/* reloc->r_offset */
-	lwz	r0,8(r9)	/* reloc->r_addend */
-	add	r0,r0,r11
-	stwx	r0,r11,r12
-	addi	r9,r9,12
-	bdnz	2b
-
-	/* Do a cache flush for our text, in case the loader didn't */
-3:	lwz	r9,p_start-p_base(r10)	/* note: these are relocated now */
-	lwz	r8,p_etext-p_base(r10)
-4:	dcbf	r0,r9
-	icbi	r0,r9
-	addi	r9,r9,0x20
-	cmplw	cr0,r9,r8
-	blt	4b
-	sync
-	isync
-
-	/* Clear the BSS */
-	lwz	r9,p_bss_start-p_base(r10)
-	lwz	r8,p_end-p_base(r10)
-	li	r0,0
-5:	stw	r0,0(r9)
-	addi	r9,r9,4
-	cmplw	cr0,r9,r8
-	blt	5b
-
-	/* Possibly set up a custom stack */
-	lwz	r8,p_pstack-p_base(r10)
-	cmpwi	r8,0
-	beq	6f
-	lwz	r1,0(r8)
-	li	r0,0
-	stwu	r0,-16(r1)	/* establish a stack frame */
-6:
-#else /* __powerpc64__ */
 	/* Save the prom pointer at p_prom. */
 	std	r5,(p_prom-p_base)(r10)
 
@@ -220,81 +123,8 @@ p_base:	mflr	r10		/* r10 now points to r
 	li	r0,0
 	stdu	r0,-16(r1)	/* establish a stack frame */
 6:
-#endif  /* __powerpc64__ */
 	/* Call platform_init() */
 	bl	platform_init
 
 	/* Call start */
 	b	start
-
-#ifdef __powerpc64__
-
-#define PROM_FRAME_SIZE 512
-#define SAVE_GPR(n, base)       std     n,8*(n)(base)
-#define REST_GPR(n, base)       ld      n,8*(n)(base)
-#define SAVE_2GPRS(n, base)     SAVE_GPR(n, base); SAVE_GPR(n+1, base)
-#define SAVE_4GPRS(n, base)     SAVE_2GPRS(n, base); SAVE_2GPRS(n+2, base)
-#define SAVE_8GPRS(n, base)     SAVE_4GPRS(n, base); SAVE_4GPRS(n+4, base)
-#define SAVE_10GPRS(n, base)    SAVE_8GPRS(n, base); SAVE_2GPRS(n+8, base)
-#define REST_2GPRS(n, base)     REST_GPR(n, base); REST_GPR(n+1, base)
-#define REST_4GPRS(n, base)     REST_2GPRS(n, base); REST_2GPRS(n+2, base)
-#define REST_8GPRS(n, base)     REST_4GPRS(n, base); REST_4GPRS(n+4, base)
-#define REST_10GPRS(n, base)    REST_8GPRS(n, base); REST_2GPRS(n+8, base)
-
-/* prom handles the jump into and return from firmware.  The prom args pointer
-   is loaded in r3. */
-.globl prom
-prom:
-	mflr	r0
-	std	r0,16(r1)
-	stdu	r1,-PROM_FRAME_SIZE(r1) /* Save SP and create stack space */
-
-	SAVE_GPR(2, r1)
-	SAVE_GPR(13, r1)
-	SAVE_8GPRS(14, r1)
-	SAVE_10GPRS(22, r1)
-	mfcr    r10
-	std     r10,8*32(r1)
-	mfmsr   r10
-	std     r10,8*33(r1)
-
-	/* remove MSR_LE from msr but keep MSR_SF */
-	mfmsr	r10
-	rldicr	r10,r10,0,62
-	mtsrr1	r10
-
-	/* Load FW address, set LR to label 1, and jump to FW */
-	bl	0f
-0:	mflr	r10
-	addi	r11,r10,(1f-0b)
-	mtlr	r11
-
-	ld	r10,(p_prom-0b)(r10)
-	mtsrr0	r10
-
-	rfid
-
-1:	/* Return from OF */
-	FIXUP_ENDIAN
-
-	/* Restore registers and return. */
-	rldicl  r1,r1,0,32
-
-	/* Restore the MSR (back to 64 bits) */
-	ld      r10,8*(33)(r1)
-	mtmsr	r10
-	isync
-
-	/* Restore other registers */
-	REST_GPR(2, r1)
-	REST_GPR(13, r1)
-	REST_8GPRS(14, r1)
-	REST_10GPRS(22, r1)
-	ld      r10,8*32(r1)
-	mtcr	r10
-
-	addi    r1,r1,PROM_FRAME_SIZE
-	ld      r0,16(r1)
-	mtlr    r0
-	blr
-#endif
diff -rupN linux-denx-3.18.25.ori/arch/powerpc/kvm/Makefile linux-denx-3.18.25.mod/arch/powerpc/kvm/Makefile
--- linux-denx-3.18.25.ori/arch/powerpc/kvm/Makefile	2016-11-17 13:38:47.915045000 +0000
+++ linux-denx-3.18.25.mod/arch/powerpc/kvm/Makefile	2017-07-11 16:38:05.778712695 +0100
@@ -2,7 +2,7 @@
 # Makefile for Kernel-based Virtual Machine module
 #
 
-subdir-ccflags-$(CONFIG_PPC_WERROR) := -Werror
+subdir-ccflags-$(CONFIG_PPC_WERROR) := -Wall
 
 ccflags-y := -Ivirt/kvm -Iarch/powerpc/kvm
 KVM := ../../../virt/kvm
diff -rupN linux-denx-3.18.25.ori/arch/powerpc/platforms/83xx/Kconfig linux-denx-3.18.25.mod/arch/powerpc/platforms/83xx/Kconfig
--- linux-denx-3.18.25.ori/arch/powerpc/platforms/83xx/Kconfig	2016-11-17 13:38:47.915045000 +0000
+++ linux-denx-3.18.25.mod/arch/powerpc/platforms/83xx/Kconfig	2017-07-11 16:38:05.785712633 +0100
@@ -50,6 +50,13 @@ config MPC834x_MDS
 	  3 PCI slots.  The PIBs PCI initialization is the bootloader's
 	  responsibility.
 
+config OLT7_8CH
+        bool "PTIN OLT7 8CH"
+        select DEFAULT_UIMAGE
+        select PPC_MPC834x
+        help
+          This option enables support for the PTIN OLT7 8CH
+
 config MPC834x_ITX
 	bool "Freescale MPC834x ITX"
 	select DEFAULT_UIMAGE
diff -rupN linux-denx-3.18.25.ori/arch/powerpc/platforms/83xx/Makefile linux-denx-3.18.25.mod/arch/powerpc/platforms/83xx/Makefile
--- linux-denx-3.18.25.ori/arch/powerpc/platforms/83xx/Makefile	2016-11-17 13:38:47.915045000 +0000
+++ linux-denx-3.18.25.mod/arch/powerpc/platforms/83xx/Makefile	2017-07-11 16:38:05.790712658 +0100
@@ -17,3 +17,4 @@ obj-$(CONFIG_SBC834x)		+= sbc834x.o
 obj-$(CONFIG_MPC837x_RDB)	+= mpc837x_rdb.o
 obj-$(CONFIG_ASP834x)		+= asp834x.o
 obj-$(CONFIG_KMETER1)		+= km83xx.o
+obj-$(CONFIG_OLT7_8CH)          += olt7_8ch.o
diff -rupN linux-denx-3.18.25.ori/arch/powerpc/platforms/83xx/olt7_8ch.c linux-denx-3.18.25.mod/arch/powerpc/platforms/83xx/olt7_8ch.c
--- linux-denx-3.18.25.ori/arch/powerpc/platforms/83xx/olt7_8ch.c	1970-01-01 01:00:00.000000000 +0100
+++ linux-denx-3.18.25.mod/arch/powerpc/platforms/83xx/olt7_8ch.c	2017-07-11 16:38:05.798712699 +0100
@@ -0,0 +1,289 @@
+/*
+ * arch/powerpc/platforms/83xx/mpc8313_rdb.c
+ *
+ * Description: MPC8313x RDB board specific routines. This file is based on mpc834x_sys.c
+ * Author: Lo Wlison <r43300@freescale.com>
+ *
+ * Copyright (C) 2006 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute  it and/or modify it
+ * under  the terms of  the GNU General  Public License as published by the
+ * Free Software Foundation;  either version 2 of the  License, or (at your
+ * option) any later version.
+ */
+
+#include <linux/stddef.h>
+#include <linux/kernel.h>
+#include <linux/init.h>
+#include <linux/errno.h>
+#include <linux/reboot.h>
+#include <linux/pci.h>
+#include <linux/kdev_t.h>
+#include <linux/major.h>
+#include <linux/console.h>
+#include <linux/delay.h>
+#include <linux/seq_file.h>
+#include <linux/root_dev.h>
+#include <linux/fsl_devices.h>
+#include <linux/rtc.h>
+
+//#include <asm/system.h>
+#include <asm/atomic.h>
+#include <asm/time.h>
+#include <asm/io.h>
+#include <asm/machdep.h>
+#include <asm/ipic.h>
+//#include <asm/bootinfo.h>
+#include <asm/irq.h>
+#include <asm/prom.h>
+#include <asm/udbg.h>
+#include <sysdev/fsl_soc.h>
+#include <sysdev/fsl_pci.h>
+
+#include "mpc83xx.h"
+#include <linux/of_platform.h>
+
+#ifdef CONFIG_GFAR_PTP_VCO
+#define MPC8313_SICRH_1588_EXT_CLK	0x00000040
+#endif
+
+#ifndef CONFIG_PCI
+unsigned long isa_io_base = 0;
+unsigned long isa_mem_base = 0;
+#endif
+
+extern enum fsl_usb2_phy_modes determine_usb_phy(const char *phy_type);
+
+static int __init olt7_8ch_usb_cfg(void)
+{
+//	u32 temp;
+//	void __iomem *immap, *usb_regs;
+//	struct device_node *np = NULL;
+//	enum fsl_usb2_phy_modes phy_mode;
+//	const void *prop;
+//	struct resource res;
+//	int ret = 0;
+//
+//	if ((np = of_find_compatible_node(NULL, "usb", "fsl-usb2-dr")) == NULL)
+//		return -ENODEV;
+//	prop = of_get_property(np, "phy_type", NULL);
+//	phy_mode = determine_usb_phy(prop);
+//
+//	/* Map IMMR space for pin and clock settings */
+//	immap = ioremap(get_immrbase(), 0x1000);
+//	if (!immap) {
+//		of_node_put(np);
+//		return -ENOMEM;
+//	}
+//
+//	/* Configure clock */
+//	temp = in_be32(immap + MPC83XX_SCCR_OFFS);
+//	temp &= ~MPC83XX_SCCR_USB_MASK;
+//	temp |= MPC83XX_SCCR_USB_DRCM_11;  /* 1:3 */
+//	out_be32(immap + MPC83XX_SCCR_OFFS, temp);
+//
+//	/* Configure pin mux for ULPI.  There is no pin mux for UTMI */
+//	if (phy_mode == FSL_USB2_PHY_ULPI) {
+//		temp = in_be32(immap + MPC83XX_SICRL_OFFS);
+//		temp &= ~MPC8313_SICRL_MASK;
+//		temp |= MPC8313_SICRL_ULPI;
+//		out_be32(immap + MPC83XX_SICRL_OFFS, temp);
+//		temp = in_be32(immap + MPC83XX_SICRH_OFFS);
+//		temp &= ~MPC8313_SICRH_MASK;
+//		temp |= MPC8313_SICRH_ULPI;
+//		out_be32(immap + MPC83XX_SICRH_OFFS, temp);
+//	}
+//
+//	iounmap(immap);
+//
+//	/* Map USB SOC space */
+//	ret = of_address_to_resource(np, 0, &res);
+//	if (ret) {
+//		of_node_put(np);
+//		return ret;
+//	}
+//	usb_regs = ioremap(res.start, res.end - res.start + 1);
+//
+//	/* Using on-chip PHY */
+//	if ((phy_mode == FSL_USB2_PHY_UTMI_WIDE) ||
+//			(phy_mode == FSL_USB2_PHY_UTMI)) {
+//		/* Set UTMI_PHY_EN, REFSEL to 48MHZ */
+//		out_be32(usb_regs + FSL_USB2_CONTROL_OFFS, 0x280);
+//	/* Using external UPLI PHY */
+//	} else if (phy_mode == FSL_USB2_PHY_ULPI) {
+//		/* Set PHY_CLK_SEL to ULPI */
+//		temp = 0x400;
+//#ifdef CONFIG_USB_OTG
+//		/* Set OTG_PORT */
+//		temp |= 0x20;
+//#endif /* CONFIG_USB_OTG */
+//		out_be32(usb_regs + FSL_USB2_CONTROL_OFFS, temp);
+//	} else {
+//		printk(KERN_ERR "USB PHY type not supported\n");
+//		ret = -EINVAL;
+//	}
+//
+//	iounmap(usb_regs);
+//	of_node_put(np);
+//	return ret;
+	return 0;
+}
+
+/* ************************************************************************
+ *
+ * Setup the architecture
+ *
+ */
+static void __init olt7_8ch_setup_arch(void)
+{
+	struct device_node *np;
+#ifdef CONFIG_GFAR_PTP_VCO
+	void __iomem *immap;
+	unsigned long sicrh;
+#endif
+
+	if (ppc_md.progress)
+		ppc_md.progress("olt7_8ch_setup_arch()", 0);
+
+	np = of_find_node_by_type(NULL, "cpu");
+	if (np != 0) {
+		const unsigned int *fp =
+			of_get_property(np, "clock-frequency", NULL);
+		if (fp != 0)
+			loops_per_jiffy = *fp / HZ;
+		else
+			loops_per_jiffy = 50000000 / HZ;
+		of_node_put(np);
+	}
+#ifdef CONFIG_FSL_PCI
+        for_each_compatible_node(np, "pci", "fsl,mpc8349-pci")
+                mpc83xx_add_bridge(np);
+
+//SMARQUES
+//void pci_process_bridge_OF_ranges(struct pci_controller *hose,
+//                                  struct device_node *dev, int primary)
+//{
+
+#endif
+
+	olt7_8ch_usb_cfg();
+
+#ifdef  CONFIG_ROOT_NFS
+	ROOT_DEV = Root_NFS;
+#else
+	ROOT_DEV = Root_HDA1;
+#endif
+
+#ifdef CONFIG_GFAR_PTP_VCO
+	immap = ioremap(get_immrbase(), 0x1000);
+	sicrh = in_be32(immap + MPC83XX_SICRH_OFFS);
+	sicrh |= MPC8313_SICRH_1588_EXT_CLK;
+	out_be32(immap + MPC83XX_SICRH_OFFS, sicrh);
+#endif
+	
+}
+
+void __init olt7_8ch_init_IRQ(void)
+{
+	struct device_node *np;
+
+	np = of_find_node_by_type(NULL, "ipic");
+	if (!np)
+		return;
+
+	ipic_init(np, 0);
+
+	of_node_put(np);
+
+	/* Initialize the default interrupt mapping priorities,
+	 * in case the boot rom changed something on us.
+	 */
+	ipic_set_default_priority();
+}
+
+/*
+ * Called very early, MMU is off, device-tree isn't unflattened
+ */
+static int __init olt7_8ch_probe(void)
+{
+	/* We always match for now, eventually we should look at the flat
+	   dev tree to ensure this is the board we are suppose to run on
+	*/
+	return 1;
+}
+
+static struct of_device_id olt7_8ch_soc_ids[] = {
+	{ .type = "soc", },
+	{},
+};
+
+static int __init olt7_8ch_declare_of_platform_devices(void)
+{
+	if (!machine_is(olt7_8ch))
+		return 0;
+
+	of_platform_bus_probe(NULL, olt7_8ch_soc_ids, NULL);
+		return 0;
+}
+
+//device_initcall(olt7_8ch_declare_of_platform_devices);
+machine_device_initcall(olt7_8ch, olt7_8ch_declare_of_platform_devices);
+
+#if defined(CONFIG_I2C_MPC) && defined(CONFIG_SENSORS_DS1337)
+extern int ds1337_do_command(int bus, int cmd, void *arg);
+extern spinlock_t rtc_lock;
+#define DS1337_GET_DATE 0
+#define DS1337_SET_DATE 1
+static void ds1337_get_rtc_time(struct rtc_time *tm)
+{
+	int result;
+	unsigned long flags;
+
+	spin_lock_irqsave(&rtc_lock, flags);
+	result = ds1337_do_command(0, DS1337_GET_DATE, tm);
+	spin_unlock_irqrestore(&rtc_lock, flags);
+	return;
+}
+
+static int ds1337_set_rtc_time(struct rtc_time *tm)
+{
+	int result;
+	unsigned long flags;
+
+	spin_lock_irqsave(&rtc_lock, flags);
+	result = ds1337_do_command(0, DS1337_SET_DATE, tm);
+	spin_unlock_irqrestore(&rtc_lock, flags);
+
+	return result;
+}
+static int __init mpc8313_rtc_hookup(void)
+{
+	struct rtc_time tm;
+	struct timespec tv;
+
+	if (!machine_is(olt7_8ch))
+		return 0;
+	ppc_md.get_rtc_time = ds1337_get_rtc_time;
+	ppc_md.set_rtc_time = ds1337_set_rtc_time;
+
+	ppc_md.get_rtc_time(&tm);
+	tv.tv_nsec = 0;
+	tv.tv_sec = mktime(tm.tm_year + 1900, tm.tm_mon + 1, tm.tm_mday,
+			tm.tm_hour, tm.tm_min, tm.tm_sec);
+	do_settimeofday(&tv);
+	return 0;
+}
+late_initcall(mpc8313_rtc_hookup);
+#endif
+
+define_machine(olt7_8ch) {
+	.name			= "OLT7_8CH",
+	.probe			= olt7_8ch_probe,
+	.setup_arch		= olt7_8ch_setup_arch,
+	.init_IRQ		= olt7_8ch_init_IRQ,
+	.get_irq		= ipic_get_irq,
+	.restart		= mpc83xx_restart,
+	.time_init		= mpc83xx_time_init,
+	.calibrate_decr		= generic_calibrate_decr,
+	.progress		= udbg_progress,
+};
diff -rupN linux-denx-3.18.25.ori/arch/powerpc/platforms/83xx/olt7_8ch.h linux-denx-3.18.25.mod/arch/powerpc/platforms/83xx/olt7_8ch.h
--- linux-denx-3.18.25.ori/arch/powerpc/platforms/83xx/olt7_8ch.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-denx-3.18.25.mod/arch/powerpc/platforms/83xx/olt7_8ch.h	2017-07-11 16:38:05.807712743 +0100
@@ -0,0 +1,24 @@
+/*
+ * arch/powerpc/platforms/83xx/mpc8313_rdb.h
+ *
+ * Description: MPC8313 RDB common board definitions. This file is based on mpc834x_sys.h
+ * Author: Lo Wlison <r43300@freescale.com>
+ *
+ * Copyright (C) 2006 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute  it and/or modify it
+ * under  the terms of  the GNU General  Public License as published by the
+ * Free Software Foundation;  either version 2 of the  License, or (at your
+ * option) any later version.
+ *
+ */
+                                                                                                                                                             
+#ifndef	__MACH_OLT7_8CH_H__
+#define	__MACH_OLT7_8CH_H__
+
+#define PIRQA	MPC83xx_IRQ_EXT1
+#define PIRQB	MPC83xx_IRQ_EXT2
+#define PIRQC	MPC83xx_IRQ_EXT1
+#define PIRQD	MPC83xx_IRQ_EXT2
+#endif				/* __MACH_OLT7_8CH_H__ */
+
diff -rupN linux-denx-3.18.25.ori/include/asm-generic/vmlinux.lds.h linux-denx-3.18.25.mod/include/asm-generic/vmlinux.lds.h
--- linux-denx-3.18.25.ori/include/asm-generic/vmlinux.lds.h	2016-11-17 14:32:01.973910000 +0000
+++ linux-denx-3.18.25.mod/include/asm-generic/vmlinux.lds.h	2017-07-11 16:38:05.823712818 +0100
@@ -629,6 +629,7 @@
 		*(.initcall##level##.init)				\
 		*(.initcall##level##s.init)				\
 
+/*
 #define INIT_CALLS							\
 		VMLINUX_SYMBOL(__initcall_start) = .;			\
 		*(.initcallearly.init)					\
@@ -643,6 +644,54 @@
 		INIT_CALLS_LEVEL(7)					\
 		VMLINUX_SYMBOL(__initcall_end) = .;
 
+#define INIT_CALLS                                                      \
+		VMLINUX_SYMBOL(__initcall_start) = .;                   \
+		*(.initcallearly.init)                                  \
+		VMLINUX_SYMBOL(__initcall0_start) = .;			\
+		*(.initcall0.init)					\
+		*(.initcall0s.init)					\
+                INIT_CALLS_LEVEL(1)                                     \
+                INIT_CALLS_LEVEL(2)                                     \
+                INIT_CALLS_LEVEL(3)                                     \
+                INIT_CALLS_LEVEL(4)                                     \
+                INIT_CALLS_LEVEL(5)                                     \
+                INIT_CALLS_LEVEL(rootfs)                                \
+                INIT_CALLS_LEVEL(6)                                     \
+                INIT_CALLS_LEVEL(7)                                     \
+                VMLINUX_SYMBOL(__initcall_end) = .;
+*/
+#define INIT_CALLS                                                       \
+	VMLINUX_SYMBOL(__initcall_start) = .;			\
+        *(.initcallearly.init)                                          \
+	VMLINUX_SYMBOL(__initcall0_start) = .;          \
+        *(.initcall0.init)                                              \
+        *(.initcall0s.init)                                             \
+	VMLINUX_SYMBOL(__initcall1_start) = .;          \
+        *(.initcall1.init)                                              \
+        *(.initcall1s.init)                                             \
+	VMLINUX_SYMBOL(__initcall2_start) = .;          \
+        *(.initcall2.init)                                              \
+        *(.initcall2s.init)                                             \
+	VMLINUX_SYMBOL(__initcall3_start) = .;          \
+        *(.initcall3.init)                                              \
+        *(.initcall3s.init)                                             \
+	VMLINUX_SYMBOL(__initcall4_start) = .;          \
+        *(.initcall4.init)                                              \
+        *(.initcall4s.init)                                             \
+	VMLINUX_SYMBOL(__initcall5_start) = .;          \
+        *(.initcall5.init)                                              \
+        *(.initcall5s.init)                                             \
+	VMLINUX_SYMBOL(__initcallrootfs_start) = .;          \
+        *(.initcallrootfs.init)                                         \
+        *(.initcallrootfss.init)                                        \
+	VMLINUX_SYMBOL(__initcall6_start) = .;          \
+        *(.initcall6.init)                                              \
+        *(.initcall6s.init)                                             \
+	VMLINUX_SYMBOL(__initcall7_start) = .;          \
+        *(.initcall7.init)                                              \
+        *(.initcall7s.init)						\
+        VMLINUX_SYMBOL(__initcall_end) = .;                       
+
 #define CON_INITCALL							\
 		VMLINUX_SYMBOL(__con_initcall_start) = .;		\
 		*(.con_initcall.init)					\
diff -rupN linux-denx-3.18.25.ori/kernel/softirq.c linux-denx-3.18.25.mod/kernel/softirq.c
--- linux-denx-3.18.25.ori/kernel/softirq.c	2016-11-17 14:41:08.453676000 +0000
+++ linux-denx-3.18.25.mod/kernel/softirq.c	2017-07-11 16:38:05.830712858 +0100
@@ -660,9 +660,9 @@ static void run_ksoftirqd(unsigned int c
 		local_irq_enable();
 		cond_resched();
 
-		preempt_disable();
+		/*preempt_disable();
 		rcu_note_context_switch(cpu);
-		preempt_enable();
+		preempt_enable();*/
 
 		return;
 	}
diff -rupN linux-denx-3.18.25.ori/tools/lib/api/Makefile linux-denx-3.18.25.mod/tools/lib/api/Makefile
--- linux-denx-3.18.25.ori/tools/lib/api/Makefile	2016-11-17 14:57:27.887009000 +0000
+++ linux-denx-3.18.25.mod/tools/lib/api/Makefile	2017-07-11 16:38:05.838712897 +0100
@@ -21,7 +21,7 @@ LIB_OBJS += $(OUTPUT)fd/array.o
 
 LIBFILE = libapikfs.a
 
-CFLAGS = -ggdb3 -Wall -Wextra -std=gnu99 -Werror -O6 -D_FORTIFY_SOURCE=2 $(EXTRA_WARNINGS) $(EXTRA_CFLAGS) -fPIC
+CFLAGS = -ggdb3 -Wall -Wextra -std=gnu99 -Wall -O6 -D_FORTIFY_SOURCE=2 $(EXTRA_WARNINGS) $(EXTRA_CFLAGS) -fPIC
 EXTLIBS = -lelf -lpthread -lrt -lm
 ALL_CFLAGS = $(CFLAGS) $(BASIC_CFLAGS) -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
 ALL_LDFLAGS = $(LDFLAGS)
diff -rupN linux-denx-3.18.25.ori/tools/perf/config/feature-checks/Makefile linux-denx-3.18.25.mod/tools/perf/config/feature-checks/Makefile
--- linux-denx-3.18.25.ori/tools/perf/config/feature-checks/Makefile	2016-11-17 14:58:32.670127000 +0000
+++ linux-denx-3.18.25.mod/tools/perf/config/feature-checks/Makefile	2017-07-11 16:38:05.847712941 +0100
@@ -42,16 +42,16 @@ BUILD = $(CC) $(CFLAGS) -o $(OUTPUT)$@ $
 ###############################
 
 test-all.bin:
-	$(BUILD) -Werror -fstack-protector-all -O2 -Werror -D_FORTIFY_SOURCE=2 -ldw -lelf -lnuma -lelf -laudit -I/usr/include/slang -lslang $(shell $(PKG_CONFIG) --libs --cflags gtk+-2.0 2>/dev/null) $(FLAGS_PERL_EMBED) $(FLAGS_PYTHON_EMBED) -DPACKAGE='"perf"' -lbfd -ldl -lz
+	$(BUILD) -Wall -fstack-protector-all -O2 -Wall -D_FORTIFY_SOURCE=2 -ldw -lelf -lnuma -lelf -laudit -I/usr/include/slang -lslang $(shell $(PKG_CONFIG) --libs --cflags gtk+-2.0 2>/dev/null) $(FLAGS_PERL_EMBED) $(FLAGS_PYTHON_EMBED) -DPACKAGE='"perf"' -lbfd -ldl -lz
 
 test-hello.bin:
 	$(BUILD)
 
 test-stackprotector-all.bin:
-	$(BUILD) -Werror -fstack-protector-all
+	$(BUILD) -Wall -fstack-protector-all
 
 test-fortify-source.bin:
-	$(BUILD) -O2 -Werror -D_FORTIFY_SOURCE=2
+	$(BUILD) -O2 -Wall -D_FORTIFY_SOURCE=2
 
 test-bionic.bin:
 	$(BUILD)
diff -rupN linux-denx-3.18.25.ori/tools/testing/selftests/powerpc/Makefile linux-denx-3.18.25.mod/tools/testing/selftests/powerpc/Makefile
--- linux-denx-3.18.25.ori/tools/testing/selftests/powerpc/Makefile	2016-11-17 14:59:03.496519000 +0000
+++ linux-denx-3.18.25.mod/tools/testing/selftests/powerpc/Makefile	2017-07-11 16:38:05.851712961 +0100
@@ -9,7 +9,7 @@ ifeq ($(ARCH),powerpc)
 GIT_VERSION = $(shell git describe --always --long --dirty || echo "unknown")
 
 CC := $(CROSS_COMPILE)$(CC)
-CFLAGS := -Wall -O2 -flto -Wall -Werror -DGIT_VERSION='"$(GIT_VERSION)"' -I$(CURDIR) $(CFLAGS)
+CFLAGS := -Wall -O2 -flto -Wall -Wall -DGIT_VERSION='"$(GIT_VERSION)"' -I$(CURDIR) $(CFLAGS)
 
 export CC CFLAGS
 
