Index: systems/linux/user/cxo2t2-arm64-3_x/Makefile
===================================================================
--- systems/linux/user/cxo2t2-arm64-3_x/Makefile	(nonexistent)
+++ systems/linux/user/cxo2t2-arm64-3_x/Makefile	(revision 3371)
@@ -0,0 +1,108 @@
+# -*- Makefile -*-
+# $Id: Makefile 1.9 Broadcom SDK $
+# $Copyright: Copyright 2012 Broadcom Corporation.
+# This program is the proprietary software of Broadcom Corporation
+# and/or its licensors, and may only be used, duplicated, modified
+# or distributed pursuant to the terms and conditions of a separate,
+# written license agreement executed between you and Broadcom
+# (an "Authorized License").  Except as set forth in an Authorized
+# License, Broadcom grants no license (express or implied), right
+# to use, or waiver of any kind with respect to the Software, and
+# Broadcom expressly reserves all rights in and to the Software
+# and all intellectual property rights therein.  IF YOU HAVE
+# NO AUTHORIZED LICENSE, THEN YOU HAVE NO RIGHT TO USE THIS SOFTWARE
+# IN ANY WAY, AND SHOULD IMMEDIATELY NOTIFY BROADCOM AND DISCONTINUE
+# ALL USE OF THE SOFTWARE.  
+#  
+# Except as expressly set forth in the Authorized License,
+#  
+# 1.     This program, including its structure, sequence and organization,
+# constitutes the valuable trade secrets of Broadcom, and you shall use
+# all reasonable efforts to protect the confidentiality thereof,
+# and to use this information only in connection with your use of
+# Broadcom integrated circuit products.
+#  
+# 2.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, THE SOFTWARE IS
+# PROVIDED "AS IS" AND WITH ALL FAULTS AND BROADCOM MAKES NO PROMISES,
+# REPRESENTATIONS OR WARRANTIES, EITHER EXPRESS, IMPLIED, STATUTORY,
+# OR OTHERWISE, WITH RESPECT TO THE SOFTWARE.  BROADCOM SPECIFICALLY
+# DISCLAIMS ANY AND ALL IMPLIED WARRANTIES OF TITLE, MERCHANTABILITY,
+# NONINFRINGEMENT, FITNESS FOR A PARTICULAR PURPOSE, LACK OF VIRUSES,
+# ACCURACY OR COMPLETENESS, QUIET ENJOYMENT, QUIET POSSESSION OR
+# CORRESPONDENCE TO DESCRIPTION. YOU ASSUME THE ENTIRE RISK ARISING
+# OUT OF USE OR PERFORMANCE OF THE SOFTWARE.
+# 
+# 3.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, IN NO EVENT SHALL
+# BROADCOM OR ITS LICENSORS BE LIABLE FOR (i) CONSEQUENTIAL,
+# INCIDENTAL, SPECIAL, INDIRECT, OR EXEMPLARY DAMAGES WHATSOEVER
+# ARISING OUT OF OR IN ANY WAY RELATING TO YOUR USE OF OR INABILITY
+# TO USE THE SOFTWARE EVEN IF BROADCOM HAS BEEN ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGES; OR (ii) ANY AMOUNT IN EXCESS OF
+# THE AMOUNT ACTUALLY PAID FOR THE SOFTWARE ITSELF OR USD 1.00,
+# WHICHEVER IS GREATER. THESE LIMITATIONS SHALL APPLY NOTWITHSTANDING
+# ANY FAILURE OF ESSENTIAL PURPOSE OF ANY LIMITED REMEDY.$
+
+#
+# This make job requires the following environment variables to be set:
+#
+# SDK                - path to StrataXGS SDK root directory
+#
+# Optionally the following environment variables can be set to
+# override the default build server configuration:
+#
+# PPC_TOOLS_DIR      - path to build tools (if not in PATH already)
+# PPC_CROSS_COMPILE  - cross compile tools prefix
+# LINUX_INCLUDE      - path to Linux kernel include directory
+#
+
+SDK :=$(shell if [ -n "$$SDK" ] ; then\
+	echo $$SDK;\
+	else\
+	cd $(dir $(lastword $(MAKEFILE_LIST))); while /usr/bin/test ! -e RELEASE ; do \
+	dir=`cd ../;pwd`;			\
+	if [ "$$dir" = "/" ] ; then		\
+	   echo Cannot find SDK in $(lastword $(MAKEFILE_LIST)) 1>&2;		\
+	   exit 1;				\
+	fi ;					\
+	cd $$dir;				\
+	done ;					\
+	pwd;					\
+	fi)
+
+ifeq ($(SDK),)					
+$(error Please run this in a tree)		
+endif 				
+
+export SDK
+
+override kernel_version=3_x
+platform=cxo2t2-arm64-$(kernel_version)
+LINUX_MAKE_USER=1
+export LINKER_RELAX=1
+
+# Select the build environment
+#BUILD_PLATFORM=ELDK
+BUILD_PLATFORM=WR_LINUX
+
+#Select the LINUX KERNEL VERSION
+#KERN_VER = 2.6.21.7
+#KERN_VER = 2.6.24.4
+KERN_VER = 4.1.8
+
+
+# Select the ELDK version
+#ELDK_VERSION=4.1
+#ELDK_VERSION=4.0
+
+#Select WRS Linux version
+WRS_LINUX_VERSION=2.0
+#WRS_LINUX_VERSION=3.0
+
+
+export KERN_VER
+export ELDK_VERSION
+export BUILD_PLATFORM
+export WRS_LINUX_VERSION
+
+include ${SDK}/make/Make.linux
+
Index: systems/linux/user/cxo2t4-e500mc-3_x/Makefile
===================================================================
--- systems/linux/user/cxo2t4-e500mc-3_x/Makefile	(nonexistent)
+++ systems/linux/user/cxo2t4-e500mc-3_x/Makefile	(revision 3371)
@@ -0,0 +1,108 @@
+# -*- Makefile -*-
+# $Id: Makefile 1.9 Broadcom SDK $
+# $Copyright: Copyright 2012 Broadcom Corporation.
+# This program is the proprietary software of Broadcom Corporation
+# and/or its licensors, and may only be used, duplicated, modified
+# or distributed pursuant to the terms and conditions of a separate,
+# written license agreement executed between you and Broadcom
+# (an "Authorized License").  Except as set forth in an Authorized
+# License, Broadcom grants no license (express or implied), right
+# to use, or waiver of any kind with respect to the Software, and
+# Broadcom expressly reserves all rights in and to the Software
+# and all intellectual property rights therein.  IF YOU HAVE
+# NO AUTHORIZED LICENSE, THEN YOU HAVE NO RIGHT TO USE THIS SOFTWARE
+# IN ANY WAY, AND SHOULD IMMEDIATELY NOTIFY BROADCOM AND DISCONTINUE
+# ALL USE OF THE SOFTWARE.  
+#  
+# Except as expressly set forth in the Authorized License,
+#  
+# 1.     This program, including its structure, sequence and organization,
+# constitutes the valuable trade secrets of Broadcom, and you shall use
+# all reasonable efforts to protect the confidentiality thereof,
+# and to use this information only in connection with your use of
+# Broadcom integrated circuit products.
+#  
+# 2.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, THE SOFTWARE IS
+# PROVIDED "AS IS" AND WITH ALL FAULTS AND BROADCOM MAKES NO PROMISES,
+# REPRESENTATIONS OR WARRANTIES, EITHER EXPRESS, IMPLIED, STATUTORY,
+# OR OTHERWISE, WITH RESPECT TO THE SOFTWARE.  BROADCOM SPECIFICALLY
+# DISCLAIMS ANY AND ALL IMPLIED WARRANTIES OF TITLE, MERCHANTABILITY,
+# NONINFRINGEMENT, FITNESS FOR A PARTICULAR PURPOSE, LACK OF VIRUSES,
+# ACCURACY OR COMPLETENESS, QUIET ENJOYMENT, QUIET POSSESSION OR
+# CORRESPONDENCE TO DESCRIPTION. YOU ASSUME THE ENTIRE RISK ARISING
+# OUT OF USE OR PERFORMANCE OF THE SOFTWARE.
+# 
+# 3.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, IN NO EVENT SHALL
+# BROADCOM OR ITS LICENSORS BE LIABLE FOR (i) CONSEQUENTIAL,
+# INCIDENTAL, SPECIAL, INDIRECT, OR EXEMPLARY DAMAGES WHATSOEVER
+# ARISING OUT OF OR IN ANY WAY RELATING TO YOUR USE OF OR INABILITY
+# TO USE THE SOFTWARE EVEN IF BROADCOM HAS BEEN ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGES; OR (ii) ANY AMOUNT IN EXCESS OF
+# THE AMOUNT ACTUALLY PAID FOR THE SOFTWARE ITSELF OR USD 1.00,
+# WHICHEVER IS GREATER. THESE LIMITATIONS SHALL APPLY NOTWITHSTANDING
+# ANY FAILURE OF ESSENTIAL PURPOSE OF ANY LIMITED REMEDY.$
+
+#
+# This make job requires the following environment variables to be set:
+#
+# SDK                - path to StrataXGS SDK root directory
+#
+# Optionally the following environment variables can be set to
+# override the default build server configuration:
+#
+# PPC_TOOLS_DIR      - path to build tools (if not in PATH already)
+# PPC_CROSS_COMPILE  - cross compile tools prefix
+# LINUX_INCLUDE      - path to Linux kernel include directory
+#
+
+SDK :=$(shell if [ -n "$$SDK" ] ; then\
+	echo $$SDK;\
+	else\
+	cd $(dir $(lastword $(MAKEFILE_LIST))); while /usr/bin/test ! -e RELEASE ; do \
+	dir=`cd ../;pwd`;			\
+	if [ "$$dir" = "/" ] ; then		\
+	   echo Cannot find SDK in $(lastword $(MAKEFILE_LIST)) 1>&2;		\
+	   exit 1;				\
+	fi ;					\
+	cd $$dir;				\
+	done ;					\
+	pwd;					\
+	fi)
+
+ifeq ($(SDK),)					
+$(error Please run this in a tree)		
+endif 				
+
+export SDK
+
+override kernel_version=3_x
+platform=cxo2t4-e500mc-$(kernel_version)
+LINUX_MAKE_USER=1
+export LINKER_RELAX=1
+
+# Select the build environment
+#BUILD_PLATFORM=ELDK
+BUILD_PLATFORM=WR_LINUX
+
+#Select the LINUX KERNEL VERSION
+#KERN_VER = 2.6.21.7
+#KERN_VER = 2.6.24.4
+KERN_VER = 4.1.8
+
+
+# Select the ELDK version
+#ELDK_VERSION=4.1
+#ELDK_VERSION=4.0
+
+#Select WRS Linux version
+WRS_LINUX_VERSION=2.0
+#WRS_LINUX_VERSION=3.0
+
+
+export KERN_VER
+export ELDK_VERSION
+export BUILD_PLATFORM
+export WRS_LINUX_VERSION
+
+include ${SDK}/make/Make.linux
+
Index: systems/linux/user/fastpath_common/Makefile.fastpath
===================================================================
--- systems/linux/user/fastpath_common/Makefile.fastpath	(nonexistent)
+++ systems/linux/user/fastpath_common/Makefile.fastpath	(revision 3371)
@@ -0,0 +1,17 @@
+#$Id: Makefile,v 1.3.4.1 Broadcom SDK $
+# $Copyright: Copyright 2007, Broadcom Corporation All Rights Reserved.
+# THIS SOFTWARE IS OFFERED "AS IS", AND BROADCOM GRANTS NO WARRANTIES
+# OF ANY KIND, EXPRESS OR IMPLIED, BY STATUTE, COMMUNICATION OR OTHERWISE.
+# BROADCOM SPECIFICALLY DISCLAIMS ANY IMPLIED WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A SPECIFIC PURPOSE OR NONINFRINGEMENT CONCERNING THIS SOFTWARE.$
+#
+LOCALDIR = systems/linux/user/fastpath_common
+
+include ${SDK}/make/Make.config
+
+lib = liblusrcmmsys
+
+include ${SDK}/make/Make.lib
+
+include ${SDK}/make/Make.depend
+
Index: systems/linux/user/fastpath_common/Makefile
===================================================================
--- systems/linux/user/fastpath_common/Makefile	(nonexistent)
+++ systems/linux/user/fastpath_common/Makefile	(revision 3371)
@@ -0,0 +1,307 @@
+# -*- Makefile -*-
+# $Id: Makefile,v 1.56 Broadcom SDK $
+# $Copyright: Copyright 2008 Broadcom Corporation.
+# This program is the proprietary software of Broadcom Corporation
+# and/or its licensors, and may only be used, duplicated, modified
+# or distributed pursuant to the terms and conditions of a separate,
+# written license agreement executed between you and Broadcom
+# (an "Authorized License").  Except as set forth in an Authorized
+# License, Broadcom grants no license (express or implied), right
+# to use, or waiver of any kind with respect to the Software, and
+# Broadcom expressly reserves all rights in and to the Software
+# and all intellectual property rights therein.  IF YOU HAVE
+# NO AUTHORIZED LICENSE, THEN YOU HAVE NO RIGHT TO USE THIS SOFTWARE
+# IN ANY WAY, AND SHOULD IMMEDIATELY NOTIFY BROADCOM AND DISCONTINUE
+# ALL USE OF THE SOFTWARE.  
+#  
+# Except as expressly set forth in the Authorized License,
+#  
+# 1.     This program, including its structure, sequence and organization,
+# constitutes the valuable trade secrets of Broadcom, and you shall use
+# all reasonable efforts to protect the confidentiality thereof,
+# and to use this information only in connection with your use of
+# Broadcom integrated circuit products.
+#  
+# 2.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, THE SOFTWARE IS
+# PROVIDED "AS IS" AND WITH ALL FAULTS AND BROADCOM MAKES NO PROMISES,
+# REPRESENTATIONS OR WARRANTIES, EITHER EXPRESS, IMPLIED, STATUTORY,
+# OR OTHERWISE, WITH RESPECT TO THE SOFTWARE.  BROADCOM SPECIFICALLY
+# DISCLAIMS ANY AND ALL IMPLIED WARRANTIES OF TITLE, MERCHANTABILITY,
+# NONINFRINGEMENT, FITNESS FOR A PARTICULAR PURPOSE, LACK OF VIRUSES,
+# ACCURACY OR COMPLETENESS, QUIET ENJOYMENT, QUIET POSSESSION OR
+# CORRESPONDENCE TO DESCRIPTION. YOU ASSUME THE ENTIRE RISK ARISING
+# OUT OF USE OR PERFORMANCE OF THE SOFTWARE.
+# 
+# 3.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, IN NO EVENT SHALL
+# BROADCOM OR ITS LICENSORS BE LIABLE FOR (i) CONSEQUENTIAL,
+# INCIDENTAL, SPECIAL, INDIRECT, OR EXEMPLARY DAMAGES WHATSOEVER
+# ARISING OUT OF OR IN ANY WAY RELATING TO YOUR USE OF OR INABILITY
+# TO USE THE SOFTWARE EVEN IF BROADCOM HAS BEEN ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGES; OR (ii) ANY AMOUNT IN EXCESS OF
+# THE AMOUNT ACTUALLY PAID FOR THE SOFTWARE ITSELF OR U.S. $1,
+# WHICHEVER IS GREATER. THESE LIMITATIONS SHALL APPLY NOTWITHSTANDING
+# ANY FAILURE OF ESSENTIAL PURPOSE OF ANY LIMITED REMEDY.$
+#
+
+override-target=unix-user
+
+NUM_CPUS = $(shell grep -c 'model name' /proc/cpuinfo)
+
+all: check build
+
+ifeq ($(LINUX_MAKE_SHARED_LIB),1)
+libext = so.$(SHAREDLIBVER)
+else
+libext = a
+endif
+
+check:
+ifeq (,$(SDK))
+	@echo 'The $$SDK environment variable is not set'; exit 1
+else
+ifeq (,$(platform))
+	@echo 'Internal error: platform variable is not set'; exit 1
+else
+include $(SDK)/make/Make.config
+LOCALDIR=systems/linux/user/fastpath_common
+include $(SDK)/make/Make.depend
+kernel-override=linux-$(platform)
+ifdef LVL7_FIXUP
+KERN_BLDROOT=$(L7_OUTPUT)/$(kernel-override)$(bldroot_suffix)
+DEST_DIR=$(L7_OUTPUT_KMOD)
+else
+ifdef BLDCONFIG
+KERN_BLDROOT=$(SDK)/build/$(BLDCONFIG)/$(kernel-override)$(bldroot_suffix)
+else
+KERN_BLDROOT=$(SDK)/build/$(kernel-override)$(bldroot_suffix)
+endif
+endif # end LVL7_FIXUP
+endif
+endif
+
+ifeq (,$(kernel_version))
+kernel_version=2_4
+endif
+
+ifeq ($(kernel_version),3_x)
+ADD_TO_CFLAGS += -DLKM_3_x
+KOBJ=ko
+else
+ifeq ($(kernel_version),2_6)
+ADD_TO_CFLAGS += -DLKM_2_6
+KOBJ=ko
+else
+ADD_TO_CFLAGS += -DLKM_2_4
+KOBJ=o
+endif
+endif
+
+ifneq (,$(TOOLS_DIR))
+override PATH := $(TOOLS_DIR):$(PATH)
+endif
+
+ifeq ($(DEST_DIR),)
+DEST_DIR=../$(platform)
+endif
+
+BCM = $(DEST_DIR)/bcm.$(targetplat)$(all_suffix)
+
+KERNEL_BDE=$(DEST_DIR)/linux-kernel-bde.$(KOBJ)
+USER_BDE=$(DEST_DIR)/linux-user-bde.$(KOBJ)
+
+NETSERVE = $(DEST_DIR)/netserve
+
+ifdef LVL7_FIXUP
+all_targets = kernel_modules user_libs $(KERNEL_BDE) $(USER_BDE)
+else
+all_targets = kernel_modules user_libs $(KERNEL_BDE) $(USER_BDE) $(BCM) $(NETSERVE)
+endif
+
+
+# Define targets to be built by default.
+build_targets = $(all_targets)
+
+ADD_TO_CFLAGS += -I$(SDK)/systems/linux/kernel/modules/include
+
+# Target specific
+ifndef LVL7_FIXUP
+ADD_TO_CFLAGS += -DLINUX_SAL_DMA_ALLOC_OVERRIDE
+#ADD_TO_CFLAGS += -DLINUX_SAL_USLEEP_OVERRIDE -DLINUX_SAL_SEM_OVERRIDE
+#ADD_TO_CFLAGS += -DLINUX_SAL_UDELAY_OVERRIDE
+else # Only override SEM
+ADD_TO_CFLAGS += -DLINUX_SAL_DMA_ALLOC_OVERRIDE
+ADD_TO_CFLAGS += -DLINUX_SAL_SEM_OVERRIDE
+endif
+
+# User BDE libraries
+ADD_TO_CFLAGS += -I$(SDK)/systems/bde/linux/include
+
+ifdef BUILD_PHYMOD
+ADD_TO_CFLAGS += $(SYSROOT)
+endif
+
+CFLAGS += $(ADD_TO_CFLAGS)
+
+ifdef ESW_CHIPS
+ESW_DIAG_LIBS = libdiag_esw.a 
+endif
+
+ifdef ROBO_CHIPS
+ROBO_DIAG_LIBS = libdiagrobo.a
+endif
+
+ifdef SBX_CHIPS
+SBX_DIAG_LIBS = \
+	libdiag_sbx.$(libext)
+endif
+
+ifdef DPP_CHIPS
+DPP_DIAG_LIBS = \
+        libdiag_dpp.$(libext)
+endif
+
+ifdef DFE_CHIPS
+DPP_DIAG_LIBS = \
+        libdiag_dfe.$(libext)
+endif
+
+DIAG_LIBS = \
+        libdiag.a       \
+        ${ESW_DIAG_LIBS} \
+        ${ROBO_DIAG_LIBS} \
+        ${SBX_DIAG_LIBS} \
+	$(COND_LIB_EAV_APPL) \
+	${COND_LIB_BCMX_DIAG} \
+        ${COND_LIB_TEST} \
+	$(COND_LIB_TCL) \
+	$(APPL_LIBS) \
+	$(COND_LIB_EDITLINE) \
+	$(COND_LIB_WBDM) \
+	$(COND_LIB_SMB) \
+	$(COND_LIB_PHY_DIAGNOSTICS) \
+	$(COND_LIB_BSAFE) \
+	libsal_appl.a libsal_appl_plat.a liblubde.a
+
+ACL_LIB = $(COND_LIB_ACL)
+BCM_LIBS = $(DIAG_LIBS) $(DRV_LIBS) $(ACL_LIB) $(COND_LIB_ETH_DRV)
+BCM_LIBS_BLD = $(addprefix $(BLDDIR)/,$(BCM_LIBS))
+
+BCM_LIBS_BLD += ${EXTERNAL_LIBS}
+ifdef LVL7_FIXUP
+  ifdef QUIET
+    entry_msg    = @${ECHO} Generating $(notdir $@)
+    OBJCOPY      := @$(OBJCOPY)
+  endif
+endif
+
+
+build:: $(BLDDIR)/.tree $(build_targets)
+
+kernel_modules:
+	$(MAKE) -j$(NUM_CPUS) -C $(SDK)/systems/bde/linux/kernel kernel_version=$(kernel_version) OPT_CFLAGS="$(ADD_TO_CFLAGS)"
+	$(MAKE) -j$(NUM_CPUS) -C $(SDK)/systems/bde/linux/user/kernel kernel_version=$(kernel_version) OPT_CFLAGS="$(ADD_TO_CFLAGS)"
+
+# LVL7_FIXUP :
+# 1. Added $(MAKE) -C $(SDK)/systems/linux/user/common -f Makefile.fastpath CFLAGS="$(CFLAGS)"
+# 2. Remove $(MAKE) -C $(SDK)/systems/drv CFLAGS="$(CFLAGS)"
+user_libs:
+	$(MAKE) -j$(NUM_CPUS) -C $(SDK)/systems/linux/user/fastpath_common -f Makefile.fastpath CFLAGS="$(CFLAGS)"
+	$(MAKE) -j$(NUM_CPUS) -C $(SDK)/systems/bde/linux/user CFLAGS="$(CFLAGS)"
+	$(MAKE) -j$(NUM_CPUS) -C $(SDK)/src CFLAGS="$(ADD_TO_CFLAGS)"
+ifdef BUILD_MACSEC
+	@${ECHO} "Building MACSEC libraries ...."
+	$(MAKE) -j$(NUM_CPUS) -C ${MACSEC_HOME} MACSEC_CC="${CC}" MACSEC_LD="${LD}" MACSEC_AR="$(AR)" MACSEC_CFLAGS="${MACSEC_BUILD_FLAG}" MACSEC_TARGET="${targetbase}/${targetplat}/${platform}"
+endif
+	$(info SYSROOT=$(SYSROOT))
+ifdef BUILD_PHYMOD
+	@${ECHO} "Building PHYMOD libraries ...."
+	$(info "CFLAGS=$(CFLAGS)")
+	$(info "ADD_TO_CFLAGS=$(ADD_TO_CFLAGS)")
+	$(MAKE) -j$(NUM_CPUS) -C $(SDK)/libs/phymod CFLAGS="$(ADD_TO_CFLAGS)"
+endif
+
+# Broadcom Command Monitor
+
+# Build platform_defines.c to compile in config.bcm
+#
+ifndef NO_SAL_APPL
+PLATFORM_DEFINES_OBJ=${BLDDIR}/platform_defines.o
+${BLDDIR}/platform_defines.c: ${SDK}/rc/config.bcm
+	$(PERL) ${SDK}/tools/bcm2c_linux.pl ${SDK}/rc/config.bcm >$@
+${BLDDIR}/platform_defines.o: ${BLDDIR}/platform_defines.c
+	${CC} ${CFLAGS} -c $< -o $@
+else
+PLATFORM_DEFINES_OBJ=
+endif
+
+
+$(BCM): $(BCM).dbg
+	$(OBJCOPY) --strip-debug $< $@
+
+
+$(BCM).dbg: $(BLDDIR)/bdecreate.o $(BLDDIR)/version.o \
+	${PLATFORM_DEFINES_OBJ} $(BCM_LIBS_BLD) 
+	@rm -f $@	# Prevent core dump if currently running
+	$(CC) $(CFLAGS) -o $@ $(BLDDIR)/bdecreate.o $(BLDDIR)/version.o \
+	${PLATFORM_DEFINES_OBJ} $(BCM_LIBS_BLD) $(LDFLAGS)
+
+$(KERNEL_BDE): $(KERN_BLDROOT)/linux-kernel-bde.$(KOBJ)
+ifdef LVL7_FIXUP
+	$(entry_msg)
+endif
+	$(OBJCOPY) --strip-debug $< $@
+
+$(USER_BDE): $(KERN_BLDROOT)/linux-user-bde.$(KOBJ)
+ifdef LVL7_FIXUP
+	$(entry_msg)
+endif
+	$(OBJCOPY) --strip-debug $< $@
+
+$(NETSERVE): $(BLDDIR)/netserve.o
+	$(CC) $(CFLAGS) -o $@ $(BLDDIR)/netserve.o $(LDFLAGS) -lutil
+	$(OBJCOPY) --strip-debug $@
+
+
+
+
+# Build Version
+
+$(BLDDIR)/version.c: $(BLDDIR)/bdecreate.o $(BCM_LIBS_BLD)
+	$(PERL) $(SDK)/tools/version.pl > $@
+$(BLDDIR)/version.o: $(BLDDIR)/version.c
+	$(CC) $(CFLAGS) -c $< -o $@
+
+clean::
+	$(MAKE) -C $(SDK)/systems/bde/linux/kernel $@
+	$(MAKE) -C $(SDK)/systems/bde/linux/user/kernel $@
+	$(MAKE) -C $(SDK)/systems/bde/linux/user CFLAGS="$(CFLAGS)" $@
+	$(MAKE) -C $(SDK)/src CFLAGS="$(CFLAGS)" $@
+	$(RM) $(BLDDIR)/version.c $(BLDDIR)/version.o
+	$(RM) $(BLDDIR)/platform_defines.c $(BLDDIR)/platform_defines.o
+	$(RM) $(BLDDIR)/bdecreate.o
+	$(RM) $(KERNEL_BDE) $(USER_BDE) $(BCM) $(BCM).dbg
+	$(RM) $(BLDDIR)/netserve.o
+	$(RM) $(NETSERVE)
+
+distclean:: clean
+
+.PHONY: variable
+
+#
+# Echo variable values used for configuration
+# usage: make VAR=CC variable
+#
+variable::
+	@echo $($(VAR))
+
+# Make.config defines remaining phony targets
+.PHONY: build kernel_modules user_libs
+
+# Output the CFLAGS used by the SDK
+#
+ifdef LVL7_FIXUP
+.PHONY: get_sdk_flags
+
+get_sdk_flags::
+	@echo "CFLAGS_SDK := $(filter -D%,$(CFLAGS))" >$(L7_OUT_SDK_FLAGS)
+endif
Index: systems/linux/user/fastpath_common/bdecreate.c
===================================================================
--- systems/linux/user/fastpath_common/bdecreate.c	(nonexistent)
+++ systems/linux/user/fastpath_common/bdecreate.c	(revision 3371)
@@ -0,0 +1,148 @@
+/*
+ * $Id: socdiag.c,v 1.13 Broadcom SDK $
+ * $Copyright: Copyright 2008 Broadcom Corporation.
+ * This program is the proprietary software of Broadcom Corporation
+ * and/or its licensors, and may only be used, duplicated, modified
+ * or distributed pursuant to the terms and conditions of a separate,
+ * written license agreement executed between you and Broadcom
+ * (an "Authorized License").  Except as set forth in an Authorized
+ * License, Broadcom grants no license (express or implied), right
+ * to use, or waiver of any kind with respect to the Software, and
+ * Broadcom expressly reserves all rights in and to the Software
+ * and all intellectual property rights therein.  IF YOU HAVE
+ * NO AUTHORIZED LICENSE, THEN YOU HAVE NO RIGHT TO USE THIS SOFTWARE
+ * IN ANY WAY, AND SHOULD IMMEDIATELY NOTIFY BROADCOM AND DISCONTINUE
+ * ALL USE OF THE SOFTWARE.  
+ *  
+ * Except as expressly set forth in the Authorized License,
+ *  
+ * 1.     This program, including its structure, sequence and organization,
+ * constitutes the valuable trade secrets of Broadcom, and you shall use
+ * all reasonable efforts to protect the confidentiality thereof,
+ * and to use this information only in connection with your use of
+ * Broadcom integrated circuit products.
+ *  
+ * 2.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, THE SOFTWARE IS
+ * PROVIDED "AS IS" AND WITH ALL FAULTS AND BROADCOM MAKES NO PROMISES,
+ * REPRESENTATIONS OR WARRANTIES, EITHER EXPRESS, IMPLIED, STATUTORY,
+ * OR OTHERWISE, WITH RESPECT TO THE SOFTWARE.  BROADCOM SPECIFICALLY
+ * DISCLAIMS ANY AND ALL IMPLIED WARRANTIES OF TITLE, MERCHANTABILITY,
+ * NONINFRINGEMENT, FITNESS FOR A PARTICULAR PURPOSE, LACK OF VIRUSES,
+ * ACCURACY OR COMPLETENESS, QUIET ENJOYMENT, QUIET POSSESSION OR
+ * CORRESPONDENCE TO DESCRIPTION. YOU ASSUME THE ENTIRE RISK ARISING
+ * OUT OF USE OR PERFORMANCE OF THE SOFTWARE.
+ * 
+ * 3.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, IN NO EVENT SHALL
+ * BROADCOM OR ITS LICENSORS BE LIABLE FOR (i) CONSEQUENTIAL,
+ * INCIDENTAL, SPECIAL, INDIRECT, OR EXEMPLARY DAMAGES WHATSOEVER
+ * ARISING OUT OF OR IN ANY WAY RELATING TO YOUR USE OF OR INABILITY
+ * TO USE THE SOFTWARE EVEN IF BROADCOM HAS BEEN ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGES; OR (ii) ANY AMOUNT IN EXCESS OF
+ * THE AMOUNT ACTUALLY PAID FOR THE SOFTWARE ITSELF OR U.S. $1,
+ * WHICHEVER IS GREATER. THESE LIMITATIONS SHALL APPLY NOTWITHSTANDING
+ * ANY FAILURE OF ESSENTIAL PURPOSE OF ANY LIMITED REMEDY.$
+ *
+ * socdiag: low-level diagnostics shell for Orion (SOC) driver.
+ */
+
+#include <unistd.h>
+#include <stdlib.h>
+
+#include <sal/core/boot.h>
+#include <sal/appl/sal.h>
+#include <sal/appl/config.h>
+#include <soc/debug.h>
+
+#include <appl/diag/system.h>
+
+#include <linux-bde.h>
+
+ibde_t *bde;
+
+/* The bus properties are (currently) the only system specific
+ * settings required. 
+ * These must be defined beforehand 
+ */
+
+#ifndef SYS_BE_PIO
+#error "SYS_BE_PIO must be defined for the target platform"
+#endif
+#ifndef SYS_BE_PACKET
+#error "SYS_BE_PACKET must be defined for the target platform"
+#endif
+#ifndef SYS_BE_OTHER
+#error "SYS_BE_OTHER must be defined for the target platform"
+#endif
+
+#if !defined(SYS_BE_PIO) || !defined(SYS_BE_PACKET) || !defined(SYS_BE_OTHER)
+#error "platform bus properties not defined."
+#endif
+
+int
+bde_create(void)
+{	
+    linux_bde_bus_t bus;
+    bus.be_pio = SYS_BE_PIO;
+    bus.be_packet = SYS_BE_PACKET;
+    bus.be_other = SYS_BE_OTHER;
+    return linux_bde_create(&bus, &bde);
+}
+
+/* PTin added: application control */
+int
+bde_destroy(void)
+{	
+    return linux_bde_destroy(bde);
+}
+
+
+/*
+ * These stubs are here for legacy compatability reasons. 
+ * They are used only by the diag/test code, not the driver, 
+ * so they are really not that important. 
+ */
+typedef void* pci_dev_t;
+
+int pci_dma_putw(pci_dev_t *dev, uint32 addr, uint32 data)
+{
+    *(uint32 *) INT_TO_PTR(addr) = data;
+    return 0;
+}
+
+int pci_dma_puth(pci_dev_t *dev, uint32 addr, uint16 data)
+{
+    *(uint16 *) INT_TO_PTR(addr) = data;
+    return 0;
+}
+
+int pci_dma_putb(pci_dev_t *dev, uint32 addr, uint8 data)
+{
+    *(uint8 *) INT_TO_PTR(addr) = data;
+    return 0;
+}
+
+uint32 pci_dma_getw(pci_dev_t *dev, uint32 addr)
+{
+  uint32 data;
+  data = *(uint32 *) INT_TO_PTR(addr);
+  return data;
+}
+
+uint16 pci_dma_geth(pci_dev_t *dev, uint32 addr)
+{
+  uint16 data;
+  data = *(uint16 *) INT_TO_PTR(addr);
+  return data;
+}
+
+uint8 pci_dma_getb(pci_dev_t *dev, uint32 addr)
+{
+  uint8 data;
+  data = *(uint8 *) INT_TO_PTR(addr);
+  return data;
+}
+
+int pci_print_all(void)
+{
+  return 0;
+}
Index: systems/linux/user/fastpath-3_x/Makefile
===================================================================
--- systems/linux/user/fastpath-3_x/Makefile	(nonexistent)
+++ systems/linux/user/fastpath-3_x/Makefile	(revision 3371)
@@ -0,0 +1,27 @@
+# -*- Makefile -*-
+# $Id: Makefile,v 1.1.2.2 Broadcom SDK $
+# $Copyright: Copyright 2007, Broadcom Corporation All Rights Reserved.
+# THIS SOFTWARE IS OFFERED "AS IS", AND BROADCOM GRANTS NO WARRANTIES
+# OF ANY KIND, EXPRESS OR IMPLIED, BY STATUTE, COMMUNICATION OR OTHERWISE.
+# BROADCOM SPECIFICALLY DISCLAIMS ANY IMPLIED WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A SPECIFIC PURPOSE OR NONINFRINGEMENT CONCERNING THIS SOFTWARE.$
+
+#
+# This make job requires the following environment variables to be set:
+#
+# SDK                - path to StrataXGS SDK root directory
+#
+# Optionally the following environment variables can be set to
+# override the default build server configuration:
+#
+# PPC_TOOLS_DIR      - path to build tools (if not in PATH already)
+# PPC_CROSS_COMPILE  - cross compile tools prefix
+# LINUX_INCLUDE      - path to Linux kernel include directory
+#
+
+override kernel_version=3_x
+platform=$(BOARD)-$(L7_CPU)-$(kernel_version)
+LINUX_MAKE_USER=1
+LINUX_MAKE_DIR := $(SDK)/systems/linux/user/fastpath_common
+include ${SDK}/make/Make.linux
+
Index: systems/linux/kernel/cxo2t2-arm64-3_x/Makefile
===================================================================
--- systems/linux/kernel/cxo2t2-arm64-3_x/Makefile	(nonexistent)
+++ systems/linux/kernel/cxo2t2-arm64-3_x/Makefile	(revision 3371)
@@ -0,0 +1,104 @@
+# -*- Makefile -*-
+# $Id: Makefile 1.10 Broadcom SDK $
+# $Copyright: Copyright 2012 Broadcom Corporation.
+# This program is the proprietary software of Broadcom Corporation
+# and/or its licensors, and may only be used, duplicated, modified
+# or distributed pursuant to the terms and conditions of a separate,
+# written license agreement executed between you and Broadcom
+# (an "Authorized License").  Except as set forth in an Authorized
+# License, Broadcom grants no license (express or implied), right
+# to use, or waiver of any kind with respect to the Software, and
+# Broadcom expressly reserves all rights in and to the Software
+# and all intellectual property rights therein.  IF YOU HAVE
+# NO AUTHORIZED LICENSE, THEN YOU HAVE NO RIGHT TO USE THIS SOFTWARE
+# IN ANY WAY, AND SHOULD IMMEDIATELY NOTIFY BROADCOM AND DISCONTINUE
+# ALL USE OF THE SOFTWARE.  
+#  
+# Except as expressly set forth in the Authorized License,
+#  
+# 1.     This program, including its structure, sequence and organization,
+# constitutes the valuable trade secrets of Broadcom, and you shall use
+# all reasonable efforts to protect the confidentiality thereof,
+# and to use this information only in connection with your use of
+# Broadcom integrated circuit products.
+#  
+# 2.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, THE SOFTWARE IS
+# PROVIDED "AS IS" AND WITH ALL FAULTS AND BROADCOM MAKES NO PROMISES,
+# REPRESENTATIONS OR WARRANTIES, EITHER EXPRESS, IMPLIED, STATUTORY,
+# OR OTHERWISE, WITH RESPECT TO THE SOFTWARE.  BROADCOM SPECIFICALLY
+# DISCLAIMS ANY AND ALL IMPLIED WARRANTIES OF TITLE, MERCHANTABILITY,
+# NONINFRINGEMENT, FITNESS FOR A PARTICULAR PURPOSE, LACK OF VIRUSES,
+# ACCURACY OR COMPLETENESS, QUIET ENJOYMENT, QUIET POSSESSION OR
+# CORRESPONDENCE TO DESCRIPTION. YOU ASSUME THE ENTIRE RISK ARISING
+# OUT OF USE OR PERFORMANCE OF THE SOFTWARE.
+# 
+# 3.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, IN NO EVENT SHALL
+# BROADCOM OR ITS LICENSORS BE LIABLE FOR (i) CONSEQUENTIAL,
+# INCIDENTAL, SPECIAL, INDIRECT, OR EXEMPLARY DAMAGES WHATSOEVER
+# ARISING OUT OF OR IN ANY WAY RELATING TO YOUR USE OF OR INABILITY
+# TO USE THE SOFTWARE EVEN IF BROADCOM HAS BEEN ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGES; OR (ii) ANY AMOUNT IN EXCESS OF
+# THE AMOUNT ACTUALLY PAID FOR THE SOFTWARE ITSELF OR USD 1.00,
+# WHICHEVER IS GREATER. THESE LIMITATIONS SHALL APPLY NOTWITHSTANDING
+# ANY FAILURE OF ESSENTIAL PURPOSE OF ANY LIMITED REMEDY.$
+
+#
+# This make job requires the following environment variables to be set:
+#
+# SDK                - path to StrataXGS SDK root directory
+#
+# Optionally the following environment variables can be set to
+# override the default build server configuration:
+#
+# PPC_TOOLS_DIR     - path to build tools (if not in PATH already)
+# PPC_CROSS_COMPILE - cross compile tools prefix
+# LINUX_INCLUDE      - path to Linux kernel include directory
+#
+
+SDK :=$(shell if [ -n "$$SDK" ] ; then\
+	echo $$SDK;\
+	else\
+	cd $(dir $(lastword $(MAKEFILE_LIST))); while /usr/bin/test ! -e RELEASE ; do \
+	dir=`cd ../;pwd`;			\
+	if [ "$$dir" = "/" ] ; then		\
+	   echo Cannot find SDK in $(lastword $(MAKEFILE_LIST)) 1>&2;		\
+	   exit 1;				\
+	fi ;					\
+	cd $$dir;				\
+	done ;					\
+	pwd;					\
+	fi)
+
+ifeq ($(SDK),)					
+$(error Please run this in a tree)		
+endif 				
+
+export SDK
+
+override kernel_version=3_x
+platform=cxo2t2-arm64-$(kernel_version)
+
+# Select build platform 
+BUILD_PLATFORM=WR_LINUX
+#BUILD_PLATFORM=ELDK
+
+#Select the LINUX KERNEL VERSION 
+#KERN_VER = 2.6.21.7
+#KERN_VER = 2.6.24.4
+KERN_VER = 4.1.8
+
+#Select the ELDK version
+#ELDK_VERSION=4.1
+#ELDK_VERSION=4.0
+
+#Select WRS linux version
+WRS_LINUX_VERSION=2.0
+#WRS_LINUX_VERSION=3.0
+
+
+export KERN_VER
+export ELDK_VERSION
+export BUILD_PLATFORM
+export WRS_LINUX_VERSION
+include ${SDK}/make/Make.linux
+
Index: systems/linux/kernel/cxo2t4-e500mc-3_x/Makefile
===================================================================
--- systems/linux/kernel/cxo2t4-e500mc-3_x/Makefile	(nonexistent)
+++ systems/linux/kernel/cxo2t4-e500mc-3_x/Makefile	(revision 3371)
@@ -0,0 +1,104 @@
+# -*- Makefile -*-
+# $Id: Makefile 1.10 Broadcom SDK $
+# $Copyright: Copyright 2012 Broadcom Corporation.
+# This program is the proprietary software of Broadcom Corporation
+# and/or its licensors, and may only be used, duplicated, modified
+# or distributed pursuant to the terms and conditions of a separate,
+# written license agreement executed between you and Broadcom
+# (an "Authorized License").  Except as set forth in an Authorized
+# License, Broadcom grants no license (express or implied), right
+# to use, or waiver of any kind with respect to the Software, and
+# Broadcom expressly reserves all rights in and to the Software
+# and all intellectual property rights therein.  IF YOU HAVE
+# NO AUTHORIZED LICENSE, THEN YOU HAVE NO RIGHT TO USE THIS SOFTWARE
+# IN ANY WAY, AND SHOULD IMMEDIATELY NOTIFY BROADCOM AND DISCONTINUE
+# ALL USE OF THE SOFTWARE.  
+#  
+# Except as expressly set forth in the Authorized License,
+#  
+# 1.     This program, including its structure, sequence and organization,
+# constitutes the valuable trade secrets of Broadcom, and you shall use
+# all reasonable efforts to protect the confidentiality thereof,
+# and to use this information only in connection with your use of
+# Broadcom integrated circuit products.
+#  
+# 2.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, THE SOFTWARE IS
+# PROVIDED "AS IS" AND WITH ALL FAULTS AND BROADCOM MAKES NO PROMISES,
+# REPRESENTATIONS OR WARRANTIES, EITHER EXPRESS, IMPLIED, STATUTORY,
+# OR OTHERWISE, WITH RESPECT TO THE SOFTWARE.  BROADCOM SPECIFICALLY
+# DISCLAIMS ANY AND ALL IMPLIED WARRANTIES OF TITLE, MERCHANTABILITY,
+# NONINFRINGEMENT, FITNESS FOR A PARTICULAR PURPOSE, LACK OF VIRUSES,
+# ACCURACY OR COMPLETENESS, QUIET ENJOYMENT, QUIET POSSESSION OR
+# CORRESPONDENCE TO DESCRIPTION. YOU ASSUME THE ENTIRE RISK ARISING
+# OUT OF USE OR PERFORMANCE OF THE SOFTWARE.
+# 
+# 3.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, IN NO EVENT SHALL
+# BROADCOM OR ITS LICENSORS BE LIABLE FOR (i) CONSEQUENTIAL,
+# INCIDENTAL, SPECIAL, INDIRECT, OR EXEMPLARY DAMAGES WHATSOEVER
+# ARISING OUT OF OR IN ANY WAY RELATING TO YOUR USE OF OR INABILITY
+# TO USE THE SOFTWARE EVEN IF BROADCOM HAS BEEN ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGES; OR (ii) ANY AMOUNT IN EXCESS OF
+# THE AMOUNT ACTUALLY PAID FOR THE SOFTWARE ITSELF OR USD 1.00,
+# WHICHEVER IS GREATER. THESE LIMITATIONS SHALL APPLY NOTWITHSTANDING
+# ANY FAILURE OF ESSENTIAL PURPOSE OF ANY LIMITED REMEDY.$
+
+#
+# This make job requires the following environment variables to be set:
+#
+# SDK                - path to StrataXGS SDK root directory
+#
+# Optionally the following environment variables can be set to
+# override the default build server configuration:
+#
+# PPC_TOOLS_DIR     - path to build tools (if not in PATH already)
+# PPC_CROSS_COMPILE - cross compile tools prefix
+# LINUX_INCLUDE      - path to Linux kernel include directory
+#
+
+SDK :=$(shell if [ -n "$$SDK" ] ; then\
+	echo $$SDK;\
+	else\
+	cd $(dir $(lastword $(MAKEFILE_LIST))); while /usr/bin/test ! -e RELEASE ; do \
+	dir=`cd ../;pwd`;			\
+	if [ "$$dir" = "/" ] ; then		\
+	   echo Cannot find SDK in $(lastword $(MAKEFILE_LIST)) 1>&2;		\
+	   exit 1;				\
+	fi ;					\
+	cd $$dir;				\
+	done ;					\
+	pwd;					\
+	fi)
+
+ifeq ($(SDK),)					
+$(error Please run this in a tree)		
+endif 				
+
+export SDK
+
+override kernel_version=3_x
+platform=cxo2t4-e500mc-$(kernel_version)
+
+# Select build platform 
+BUILD_PLATFORM=WR_LINUX
+#BUILD_PLATFORM=ELDK
+
+#Select the LINUX KERNEL VERSION 
+#KERN_VER = 2.6.21.7
+#KERN_VER = 2.6.24.4
+KERN_VER = 4.1.8
+
+#Select the ELDK version
+#ELDK_VERSION=4.1
+#ELDK_VERSION=4.0
+
+#Select WRS linux version
+WRS_LINUX_VERSION=2.0
+#WRS_LINUX_VERSION=3.0
+
+
+export KERN_VER
+export ELDK_VERSION
+export BUILD_PLATFORM
+export WRS_LINUX_VERSION
+include ${SDK}/make/Make.linux
+
Index: make/Makefile.linux-kernel-3_x
===================================================================
--- make/Makefile.linux-kernel-3_x	(nonexistent)
+++ make/Makefile.linux-kernel-3_x	(revision 3371)
@@ -0,0 +1,173 @@
+# $Id: Makefile.linux-kernel-3_8,v 1.40 Broadcom SDK $
+# $Copyright: Copyright 2012 Broadcom Corporation.
+# This program is the proprietary software of Broadcom Corporation
+# and/or its licensors, and may only be used, duplicated, modified
+# or distributed pursuant to the terms and conditions of a separate,
+# written license agreement executed between you and Broadcom
+# (an "Authorized License").  Except as set forth in an Authorized
+# License, Broadcom grants no license (express or implied), right
+# to use, or waiver of any kind with respect to the Software, and
+# Broadcom expressly reserves all rights in and to the Software
+# and all intellectual property rights therein.  IF YOU HAVE
+# NO AUTHORIZED LICENSE, THEN YOU HAVE NO RIGHT TO USE THIS SOFTWARE
+# IN ANY WAY, AND SHOULD IMMEDIATELY NOTIFY BROADCOM AND DISCONTINUE
+# ALL USE OF THE SOFTWARE.  
+#  
+# Except as expressly set forth in the Authorized License,
+#  
+# 1.     This program, including its structure, sequence and organization,
+# constitutes the valuable trade secrets of Broadcom, and you shall use
+# all reasonable efforts to protect the confidentiality thereof,
+# and to use this information only in connection with your use of
+# Broadcom integrated circuit products.
+#  
+# 2.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, THE SOFTWARE IS
+# PROVIDED "AS IS" AND WITH ALL FAULTS AND BROADCOM MAKES NO PROMISES,
+# REPRESENTATIONS OR WARRANTIES, EITHER EXPRESS, IMPLIED, STATUTORY,
+# OR OTHERWISE, WITH RESPECT TO THE SOFTWARE.  BROADCOM SPECIFICALLY
+# DISCLAIMS ANY AND ALL IMPLIED WARRANTIES OF TITLE, MERCHANTABILITY,
+# NONINFRINGEMENT, FITNESS FOR A PARTICULAR PURPOSE, LACK OF VIRUSES,
+# ACCURACY OR COMPLETENESS, QUIET ENJOYMENT, QUIET POSSESSION OR
+# CORRESPONDENCE TO DESCRIPTION. YOU ASSUME THE ENTIRE RISK ARISING
+# OUT OF USE OR PERFORMANCE OF THE SOFTWARE.
+# 
+# 3.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, IN NO EVENT SHALL
+# BROADCOM OR ITS LICENSORS BE LIABLE FOR (i) CONSEQUENTIAL,
+# INCIDENTAL, SPECIAL, INDIRECT, OR EXEMPLARY DAMAGES WHATSOEVER
+# ARISING OUT OF OR IN ANY WAY RELATING TO YOUR USE OF OR INABILITY
+# TO USE THE SOFTWARE EVEN IF BROADCOM HAS BEEN ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGES; OR (ii) ANY AMOUNT IN EXCESS OF
+# THE AMOUNT ACTUALLY PAID FOR THE SOFTWARE ITSELF OR USD 1.00,
+# WHICHEVER IS GREATER. THESE LIMITATIONS SHALL APPLY NOTWITHSTANDING
+# ANY FAILURE OF ESSENTIAL PURPOSE OF ANY LIMITED REMEDY.$
+
+# Configuration Flags
+
+# Filter out features that cannot or should not be supported in kernel mode
+
+_FEATURE_EXCLUDE_LIST += EDITLINE TCL OOB_RCPU CINT APIMODE DUNE_UI C_UNIT
+FEATURE_EXCLUDE_LIST = $(sort $(_FEATURE_EXCLUDE_LIST))
+
+# Tools
+
+# Conditionally Replaces DEFAULT var
+ifeq ($(origin CC),default)
+CC	= $(LSRUN) $(CROSS_COMPILE)gcc
+endif
+
+ifeq ($(origin CXX),default)
+CXX	= $(CROSS_COMPILE)g++
+endif
+
+ifeq ($(origin LD),default)
+LD	= $(CROSS_COMPILE)ld
+endif
+
+ifeq ($(origin AR),default)
+AR	= $(CROSS_COMPILE)ar
+endif
+
+ifeq ($(origin AS),default)
+AS	= $(CROSS_COMPILE)as
+endif
+
+ifeq ($(origin ARFLAGS),default)
+	ARFLAGS	= -rc
+endif
+
+STRIP	= $(CROSS_COMPILE)strip
+RANLIB  = $(CROSS_COMPILE)ranlib
+OBJCOPY = $(CROSS_COMPILE)objcopy
+NM      = $(CROSS_COMPILE)nm
+
+# Handle differences between gcc 2.x and gcc 3.x
+gcc-tune-flag = $(shell if ${CC} -dumpspecs | grep mcpu >/dev/null; then echo cpu; else echo tune; fi)
+
+# Configuration Variables
+
+# OSType Defines: This defines the type of RTOS or microkernel which you
+# are compiling the SAL (and its associated driver) for. New platforms
+# can be created by porting the routines (system.c) to your platform and
+# adding the define in this Makefile.
+
+OSTYPE = LINUX
+
+#
+# ORIGIN is used to Optionally select different CFLAGS. It is used to import
+# source from other vendors. If SOURCE=Broadcom, then the BCM_ flags are added
+# to those passed to the compiler. If SOURCE != Broadcom, BCM_ flags are NOT
+# added.
+#
+# Default specifies Broadcom
+#
+ifndef ORIGIN
+	ORIGIN = Broadcom
+endif
+
+#
+# STD_{C|CPP|CXX}FLAGS - Standard flags used by ALL compilations
+# BCM_{C|CPP|CXX}FLAGS - Flags used for Broadcom source files
+# OPT_{C|CPP|CXX}FLAGS - Defined in local make files BEFORE inclusion of 
+#			 this Makefile, to define local "Extra" flags.
+#
+
+ifdef IPROC_BUILD
+CFGFLAGS += -I$(SDK)/systems/linux/kernel/modules/include \
+	    -I$(SDK)/systems/bde/linux/include \
+	    -I$(LINUX_INCLUDE) \
+            -I$(KERNDIR)/arch/$(ARCH)
+else
+CFGFLAGS += -I$(SDK)/systems/linux/kernel/modules/include \
+	    -I$(SDK)/systems/bde/linux/include \
+	    -I$(LINUX_INCLUDE) \
+	    -I$(LINUX_INCLUDE)/asm/gcc \
+	    -I$(LINUX_INCLUDE)/asm/mach-generic \
+            -I$(KERNDIR)/arch/$(ARCH)
+endif
+
+CFGFLAGS += -DNO_CTRL_C -DNO_MEMTUNE
+#CFGFLAGS += -DNO_FILEIO
+
+CFGFLAGS += -D$(OSTYPE)
+
+# No user sal for the linux kernel
+# NO_SAL_APPL=1
+
+STD_CFLAGS = $(KFLAGS) $(CFGFLAGS)
+
+
+STD_CPPFLAGS = ${STD_CFLAGS}
+STD_CXXFLAGS = ${STD_CFLAGS}
+
+ifndef BCM_CFLAGS
+BCM_CFLAGS   = -Wall -Werror
+endif
+
+BCM_CPPFLAGS = ${BCM_CFLAGS}
+BCM_CXXFLAGS = ${BCM_CFLAGS}
+
+ifeq (${ORIGIN}, Broadcom)
+	CFLAGS   += ${STD_CFLAGS}   ${BCM_CFLAGS}   ${OPT_CFLAGS}
+	CPPFLAGS += ${STD_CPPFLAGS} ${BCM_CPPFLAGS} ${OPT_CPPFLAGS}
+	CXXFLAGS += ${STD_CXXFLAGS} ${BCM_CXXFLAGS} ${OPT_CXXFLAGS}
+else
+	CFLAGS   += ${STD_CFLAGS}   ${OPT_CFLAGS}
+	CPPFLAGS += ${STD_CPPFLAGS} ${OPT_CPPFLAGS}
+	CXXFLAGS += ${STD_CXXFLAGS} ${OPT_CXXFLAGS}
+endif
+
+#
+# Ignore pedantic flag for kernel modules
+#
+ifdef DEBUG_PEDANTIC
+DEBUG_PEDANTIC = FALSE
+endif
+
+#
+# DEPEND is used as a command to generate the list of dependencies.
+# The format of the output must be 
+#	"file.o : file.c a/b/c.h d/e/f.h ...", 
+# if it is on multiple lines, each line must end in a backslash. 
+# The output MUST be on standard out.
+#
+DEPEND = ${CC} -M $(CFLAGS) $< 
Index: make/Makefile.linux-cxo2t2-arm64-3_x
===================================================================
--- make/Makefile.linux-cxo2t2-arm64-3_x	(nonexistent)
+++ make/Makefile.linux-cxo2t2-arm64-3_x	(revision 3371)
@@ -0,0 +1,125 @@
+# $Id: Makefile.linux-iproc-3_6 1.2 Broadcom SDK $
+# $Copyright: Copyright 2012 Broadcom Corporation.
+# This program is the proprietary software of Broadcom Corporation
+# and/or its licensors, and may only be used, duplicated, modified
+# or distributed pursuant to the terms and conditions of a separate,
+# written license agreement executed between you and Broadcom
+# (an "Authorized License").  Except as set forth in an Authorized
+# License, Broadcom grants no license (express or implied), right
+# to use, or waiver of any kind with respect to the Software, and
+# Broadcom expressly reserves all rights in and to the Software
+# and all intellectual property rights therein.  IF YOU HAVE
+# NO AUTHORIZED LICENSE, THEN YOU HAVE NO RIGHT TO USE THIS SOFTWARE
+# IN ANY WAY, AND SHOULD IMMEDIATELY NOTIFY BROADCOM AND DISCONTINUE
+# ALL USE OF THE SOFTWARE.  
+#  
+# Except as expressly set forth in the Authorized License,
+#  
+# 1.     This program, including its structure, sequence and organization,
+# constitutes the valuable trade secrets of Broadcom, and you shall use
+# all reasonable efforts to protect the confidentiality thereof,
+# and to use this information only in connection with your use of
+# Broadcom integrated circuit products.
+#  
+# 2.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, THE SOFTWARE IS
+# PROVIDED "AS IS" AND WITH ALL FAULTS AND BROADCOM MAKES NO PROMISES,
+# REPRESENTATIONS OR WARRANTIES, EITHER EXPRESS, IMPLIED, STATUTORY,
+# OR OTHERWISE, WITH RESPECT TO THE SOFTWARE.  BROADCOM SPECIFICALLY
+# DISCLAIMS ANY AND ALL IMPLIED WARRANTIES OF TITLE, MERCHANTABILITY,
+# NONINFRINGEMENT, FITNESS FOR A PARTICULAR PURPOSE, LACK OF VIRUSES,
+# ACCURACY OR COMPLETENESS, QUIET ENJOYMENT, QUIET POSSESSION OR
+# CORRESPONDENCE TO DESCRIPTION. YOU ASSUME THE ENTIRE RISK ARISING
+# OUT OF USE OR PERFORMANCE OF THE SOFTWARE.
+# 
+# 3.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, IN NO EVENT SHALL
+# BROADCOM OR ITS LICENSORS BE LIABLE FOR (i) CONSEQUENTIAL,
+# INCIDENTAL, SPECIAL, INDIRECT, OR EXEMPLARY DAMAGES WHATSOEVER
+# ARISING OUT OF OR IN ANY WAY RELATING TO YOUR USE OF OR INABILITY
+# TO USE THE SOFTWARE EVEN IF BROADCOM HAS BEEN ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGES; OR (ii) ANY AMOUNT IN EXCESS OF
+# THE AMOUNT ACTUALLY PAID FOR THE SOFTWARE ITSELF OR USD 1.00,
+# WHICHEVER IS GREATER. THESE LIMITATIONS SHALL APPLY NOTWITHSTANDING
+# ANY FAILURE OF ESSENTIAL PURPOSE OF ANY LIMITED REMEDY.$
+# Makefile for iproc-CMICd
+
+# User must select one platform from below.By default ARM_LINUX is selected. .
+ifeq (,$(BUILD_PLATFORM))
+BUILD_PLATFORM=ARM_LINUX
+endif
+
+# Toolchain base directory for iPROC-CMICd devices
+# TOOLCHAIN_BASE_DIR:= /projects/ntsw-tools/linux/iproc_ldks/ldk36/iproc
+TOOLCHAIN_BASE_DIR ?= /opt/gcc-linaro-7.2.1-2017.11-i686_aarch64-linux-gnu
+
+# Compiler for target architecture
+TARGET_ARCHITECTURE?= aarch64-linux-gnu
+
+# Kernel directory for iPROC-CMICd devices
+# KERNDIR:= $(TOOLCHAIN_BASE_DIR)/kernel/linux-custom
+KERNDIR ?= /home/mruas/svnrepo/olt-switchsw/branches/switchdrvr-4.0/lib/kernel/linux-kernel-4.9.2
+
+ifeq (,$(CROSS_COMPILE))
+CROSS_COMPILE:= $(TARGET_ARCHITECTURE)-
+endif
+
+# arm9tools
+TOOLCHAIN_BIN_DIR ?= $(TOOLCHAIN_BASE_DIR)/bin
+LD_LIBRARY_PATH ?= $(TOOLCHAIN_BASE_DIR)/lib
+override PATH:=$(TOOLCHAIN_BASE_DIR)/$(TARGET_ARCHITECTURE)/bin:$(TOOLCHAIN_BIN_DIR):$(PATH)
+
+export TOOLCHAIN_BIN_DIR LD_LIBRARY_PATH
+
+# Default Linux include directory
+ifeq (,$(LINUX_INCLUDE))
+LINUX_INCLUDE := $(KERNDIR)/include
+endif
+
+CFGFLAGS += -DSYS_BE_PIO=0 -DSYS_BE_PACKET=0 -DSYS_BE_OTHER=0
+ENDIAN = LE_HOST=1
+CFGFLAGS += -D$(ENDIAN)
+CFGFLAGS += -DBCM_PLATFORM_STRING=\"CXO2T2\"
+CFGFLAGS += -DPHYS_ADDRS_ARE_64BITS
+CFGFLAGS += -DSAL_BDE_DMA_MEM_DEFAULT=16
+CFGFLAGS += -DSAL_SPL_LOCK_ON_IRQ=1
+#CFGFLAGS += -DSAL_BDE_32BIT_USER_64BIT_KERNEL
+#CFGFLAGS += -DSAL_BDE_USE_MMAP2
+#CFGFLAGS += -DSAL_BDE_CACHE_DMA_MEM
+#CFGFLAGS += -DNONCOHERENT_DMA_MEMORY
+#CFGFLAGS += -DREMAP_DMA_NONCACHED
+CFGFLAGS += -DPTRS_ARE_64BITS
+CFGFLAGS += -DBCM_MONOTONIC_TIME
+
+ARCH = arm64
+KBUILD_VERBOSE = 1
+
+export ARCH KBUILD_VERBOSE
+
+comma = ,
+basetarget = $(basename $(notdir $@))
+modname = $(basetarget)
+
+# Extra variables.
+EXTRA_CFLAGS = -D"KBUILD_STR(s)=\#s" $(basename_flags) $(modname_flags)
+EXTRA_CFLAGS+= -Wno-unused-but-set-variable -Wno-pointer-sign
+EXTRA_CFLAGS+= -ffunction-sections -fdata-sections
+EXTRA_CFLAGS+= --sysroot=/opt/sysroot-glibc-linaro-2.25-2017.11-aarch64-linux-gnu
+
+name-fix = $(subst $(comma),_,$(subst -,_,$1))
+basename_flags = -D"KBUILD_BASENAME=KBUILD_STR($(call name-fix,$(basetarget)))"
+modname_flags  = $(if $(filter 1,$(words $(modname))),\
+                 -D"KBUILD_MODNAME=KBUILD_STR($(call name-fix,$(modname)))")
+
+KFLAG_INCLD = $(TOOLCHAIN_BASE_DIR)/lib/gcc/$(TARGET_ARCHITECTURE)/7.2.1/include
+#KFLAG_INCLD = $(TOOLCHAIN_BASE_DIR)/lib/gcc/$(TARGET_ARCHITECTURE)/4.9.4/include
+
+ifeq (,$(KFLAGS))
+#KFLAGS := -D__LINUX_ARM_ARCH__=7 -D__KERNEL__ -nostdinc  -isystem $(KFLAG_INCLD) -I$(LINUX_INCLUDE) -include $(LINUX_INCLUDE)/linux/version.h -include $(LINUX_INCLUDE)/generated/autoconf.h -I$(KERNDIR)/arch/arm/include -I$(KERNDIR)/arch/arm/include/generated -I$(KERNDIR)/arch/arm/mach-northstar/include -I$(KERNDIR)/arch/arm/plat-iproc/include -Wall -Wstrict-prototypes -Wno-trigraphs -Os -fno-strict-aliasing -fno-common -marm -mabi=aapcs-linux -fno-pic -pipe -msoft-float -ffreestanding -march=armv7-a -mfpu=vfp -mfloat-abi=softfp -fomit-frame-pointer -g -fno-stack-protector -Wdeclaration-after-statement -Wno-pointer-sign -mlong-calls
+KFLAGS := -D__LINUX_ARM_ARCH__=8 -D__KERNEL__ -nostdinc  -isystem $(KFLAG_INCLD) -I$(LINUX_INCLUDE) -I$(LINUX_INCLUDE)/linux -include $(LINUX_INCLUDE)/generated/uapi/linux/version.h -include $(LINUX_INCLUDE)/generated/autoconf.h -I$(LINUX_INCLUDE)/generated/uapi -I$(KERNDIR)/arch/$(ARCH) -I$(KERNDIR)/arch/$(ARCH)/include -I$(KERNDIR)/include/asm-generic -I$(KERNDIR)/arch/$(ARCH)/include/uapi -I$(KERNDIR)/include/uapi -I$(KERNDIR)/arch/$(ARCH)/include/generated -Wall -Wstrict-prototypes -Wno-trigraphs -O2 -fno-strict-aliasing -fno-common -fno-pic -pipe -ffreestanding -march=armv8-a -fomit-frame-pointer -g1 -fno-stack-protector -Wdeclaration-after-statement -mcmodel=large
+#-mpc-relative-literal-loads
+#-fcode-hoisting -finline-small-functions -fcrossjumping -fira-region=one -ffunction-sections -fdata-sections
+#-funwind-tables
+endif
+
+ifneq ($(targetplat),user)
+include ${SDK}/make/Makefile.linux-kernel-3_x
+endif
Index: make/Makefile.linux-cxo2t4-e500mc-3_x
===================================================================
--- make/Makefile.linux-cxo2t4-e500mc-3_x	(nonexistent)
+++ make/Makefile.linux-cxo2t4-e500mc-3_x	(revision 3371)
@@ -0,0 +1,305 @@
+# $Id: Makefile.linux-gto-2_6,v 1.28 Broadcom SDK $
+# $Copyright: Copyright 2010 Broadcom Corporation.
+# This program is the proprietary software of Broadcom Corporation
+# and/or its licensors, and may only be used, duplicated, modified
+# or distributed pursuant to the terms and conditions of a separate,
+# written license agreement executed between you and Broadcom
+# (an "Authorized License").  Except as set forth in an Authorized
+# License, Broadcom grants no license (express or implied), right
+# to use, or waiver of any kind with respect to the Software, and
+# Broadcom expressly reserves all rights in and to the Software
+# and all intellectual property rights therein.  IF YOU HAVE
+# NO AUTHORIZED LICENSE, THEN YOU HAVE NO RIGHT TO USE THIS SOFTWARE
+# IN ANY WAY, AND SHOULD IMMEDIATELY NOTIFY BROADCOM AND DISCONTINUE
+# ALL USE OF THE SOFTWARE.  
+#  
+# Except as expressly set forth in the Authorized License,
+#  
+# 1.     This program, including its structure, sequence and organization,
+# constitutes the valuable trade secrets of Broadcom, and you shall use
+# all reasonable efforts to protect the confidentiality thereof,
+# and to use this information only in connection with your use of
+# Broadcom integrated circuit products.
+#  
+# 2.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, THE SOFTWARE IS
+# PROVIDED "AS IS" AND WITH ALL FAULTS AND BROADCOM MAKES NO PROMISES,
+# REPRESENTATIONS OR WARRANTIES, EITHER EXPRESS, IMPLIED, STATUTORY,
+# OR OTHERWISE, WITH RESPECT TO THE SOFTWARE.  BROADCOM SPECIFICALLY
+# DISCLAIMS ANY AND ALL IMPLIED WARRANTIES OF TITLE, MERCHANTABILITY,
+# NONINFRINGEMENT, FITNESS FOR A PARTICULAR PURPOSE, LACK OF VIRUSES,
+# ACCURACY OR COMPLETENESS, QUIET ENJOYMENT, QUIET POSSESSION OR
+# CORRESPONDENCE TO DESCRIPTION. YOU ASSUME THE ENTIRE RISK ARISING
+# OUT OF USE OR PERFORMANCE OF THE SOFTWARE.
+# 
+# 3.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, IN NO EVENT SHALL
+# BROADCOM OR ITS LICENSORS BE LIABLE FOR (i) CONSEQUENTIAL,
+# INCIDENTAL, SPECIAL, INDIRECT, OR EXEMPLARY DAMAGES WHATSOEVER
+# ARISING OUT OF OR IN ANY WAY RELATING TO YOUR USE OF OR INABILITY
+# TO USE THE SOFTWARE EVEN IF BROADCOM HAS BEEN ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGES; OR (ii) ANY AMOUNT IN EXCESS OF
+# THE AMOUNT ACTUALLY PAID FOR THE SOFTWARE ITSELF OR USD 1.00,
+# WHICHEVER IS GREATER. THESE LIMITATIONS SHALL APPLY NOTWITHSTANDING
+# ANY FAILURE OF ESSENTIAL PURPOSE OF ANY LIMITED REMEDY.$
+
+# User must select one platform from below.By default WR_LINUX is selected. .
+ifeq (,$(BUILD_PLATFORM))
+#BUILD_PLATFORM=ELDK
+BUILD_PLATFORM=WR_LINUX
+endif
+
+# Specify the ELDK version you want to use for building SDK.
+ifeq (,$(ELDK_VERSION))
+ifeq (ELDK,$(BUILD_PLATFORM))
+ELDK_VERSION=4.0
+endif
+endif
+
+# Specify the KERNEL VERSION  you want to use for building SDK.
+ifeq (,$(KERN_VER))
+ifeq (ELDK,$(BUILD_PLATFORM))
+KERN_VER=4.1.8
+#2.6.21.7
+endif
+endif
+
+
+# Specify the Windriver Linux version here.For example '2.0' as shown below.
+ifeq (WR_LINUX,$(BUILD_PLATFORM))
+ifeq (,$(WRS_LINUX_VERSION))
+WRS_LINUX_VERSION=2.0
+#WRS_LINUX_VERSION=2.6.27
+endif
+endif
+
+
+#glibc_small and glibc_std have their own cross-compilation tools and and path for these tools are different as implemented below. To enable glibc_small build, line given below should be uncommented.
+
+#WRL_GLIBC_SMALL=TRUE
+
+ifeq (2.0,$(WRS_LINUX_VERSION))
+ifeq ($(WRL_GLIBC_SMALL),TRUE)
+
+ifeq (,$(CROSS_COMPILE))
+CROSS_COMPILE := /opt/eldk/usr/bin/ppc_85xxDP-
+endif
+
+KERNDIR ?= $(KERNEL_PATH)
+
+else
+
+ifeq (,$(CROSS_COMPILE))
+CROSS_COMPILE := /opt/eldk/usr/bin/ppc_85xxDP-
+endif
+
+KERNDIR ?= $(KERNEL_PATH)
+
+endif 
+
+export WRL_GLIBC_SMALL
+
+# After this point glibc_std and glibc_small share these flags
+
+WRLINUX_BASE ?=/tools/windriver/linux_ed/2.0_GA/Linux
+TOOLCHAIN_EXEC_PREFIX=$(WRLINUX_BASE)/gnu/4.1-wrlinux-2.0/x86-linux2
+TOOLCHAIN_BIN_DIR=$(TOOLCHAIN_EXEC_PREFIX)
+WIND_LIC_PROXY=$(WRLINUX_BASE)/setup/x86-linux2/bin
+WRLINUX_GNU_PATH = $(WRLINUX_BASE)/gnu/4.1-wrlinux-2.0/x86-linux2/bin
+override PATH := $(TOOLCHAIN_EXEC_PREFIX):$(KERNDIR)/../../host-cross/bin:$(KERNDIR)/../../host-cross/powerpc-wrs-linux-gnu/bin:$(WRLINUX_GNU_PATH):$(PATH)
+export TOOLCHAIN_EXEC_PREFIX TOOLCHAIN_BIN_DIR WIND_LIC_PROXY
+endif
+
+
+ifeq (3.0,$(WRS_LINUX_VERSION))
+ifeq ($(WRL_GLIBC_SMALL),TRUE)
+
+ifeq (,$(CROSS_COMPILE))
+CROSS_COMPILE := /opt/eldk/usr/bin/ppc_85xxDP-
+endif
+
+KERNDIR ?= /home/mruas/repositorio/svn/lib/kernel/linux-2.6.35_gcc_4.2.2
+KERNDIR_STD ?= /home/mruas/repositorio/svn/lib/kernel/linux-2.6.35_gcc_4.2.2
+
+else
+
+ifeq (,$(CROSS_COMPILE))
+CROSS_COMPILE := /opt/eldk/usr/bin/ppc_85xxDP-
+endif
+
+KERNDIR ?= $(KERNEL_PATH)
+KERNDIR_STD ?= $(KERNEL_PATH)
+
+endif 
+
+export WRL_GLIBC_SMALL
+
+# After this point glibc_std and glibc_small share these flags
+
+WRLINUX_BASE ?=/tools/windriver/linux_ed/3.0/Linux
+TOOLCHAIN_EXEC_PREFIX=$(WRLINUX_BASE)/wrlinux-3.0/layers/wrll-toolchain-4.3-85/powerpc/toolchain/x86-linux2
+TOOLCHAIN_BIN_DIR=$(TOOLCHAIN_EXEC_PREFIX)
+WIND_LIC_PROXY=$(WRLINUX_BASE)/setup/x86-linux2/bin
+WRLINUX_GNU_PATH = $(WRLINUX_BASE)/wrlinux-3.0/layers/wrll-toolchain-4.3-85/powerpc/toolchain/x86-linux2/bin
+override PATH := $(TOOLCHAIN_EXEC_PREFIX):$(KERNDIR_STD)/../../host-cross/bin:$(KERNDIR_STD)/../../host-cross/powerpc-wrs-linux-gnu/bin:$(WRLINUX_GNU_PATH):$(PATH)
+export TOOLCHAIN_EXEC_PREFIX TOOLCHAIN_BIN_DIR WIND_LIC_PROXY
+
+LINUX_INCLUDE_STD := $(KERNDIR_STD)/include
+
+endif
+
+ifeq (ELDK,$(BUILD_PLATFORM))
+ifeq (2.6.21.7, $(KERN_VER))
+
+ifeq (,$(CROSS_COMPILE))
+CROSS_COMPILE := /opt/eldk/usr/bin/ppc_85xxDP-
+endif
+
+ifeq (4.0, $(ELDK_VERSION))
+override PATH := /tools/eldk/4.0/usr/bin:$(PATH)
+else
+override PATH := /tools/eldk/4.1/usr/bin:$(PATH)
+endif
+KERNDIR ?= $(KERNEL_PATH)
+endif
+endif
+
+ifeq (ELDK,$(BUILD_PLATFORM))
+ifeq (2.6.24.4, $(KERN_VER))
+
+ifeq (,$(CROSS_COMPILE))
+CROSS_COMPILE := /opt/eldk/usr/bin/ppc_85xxDP-
+endif
+
+ifeq (4.0, $(ELDK_VERSION))
+override PATH := /tools/eldk/4.0/usr/bin:$(PATH)
+else
+override PATH := /tools/eldk/4.1/usr/bin:$(PATH)
+endif
+KERNDIR ?= $(KERNEL_PATH)
+endif
+endif
+
+
+ifeq (ELDK,$(BUILD_PLATFORM))
+ifeq (2.6.25, $(KERN_VER))
+
+ifeq (,$(CROSS_COMPILE))
+CROSS_COMPILE := /opt/eldk/usr/bin/ppc_85xxDP-
+endif
+
+ifeq (4.0, $(ELDK_VERSION))
+override PATH := /tools/eldk/4.0/usr/bin:$(PATH)
+else
+override PATH := /tools/eldk/4.1/usr/bin:$(PATH)
+endif
+KERNDIR ?= $(KERNEL_PATH)
+endif
+endif
+
+
+# Default Linux include directory
+ifeq (,$(LINUX_INCLUDE))
+LINUX_INCLUDE := $(KERNDIR)/include
+endif
+
+CFGFLAGS += -DSYS_BE_PIO=1 -DSYS_BE_PACKET=0 -DSYS_BE_OTHER=1
+ENDIAN = BE_HOST=1
+CFGFLAGS += -D$(ENDIAN) 
+CFGFLAGS += -DBCM_PLATFORM_STRING=\"OLT_P2040\"
+#CFGFLAGS += -DSAL_BDE_32BIT_USER_64BIT_KERNEL
+CFGFLAGS += -DPHYS_ADDRS_ARE_64BITS
+CFGFLAGS += -DSAL_BDE_USE_MMAP64
+#CFGFLAGS += -DSAL_BDE_USE_MMAP2
+#CFGFLAGS += -DSAL_BDE_CACHE_DMA_MEM
+CFGFLAGS += -DSAL_BDE_DMA_MEM_DEFAULT=16
+#CFGFLAGS += -DNONCOHERENT_DMA_MEMORY
+#CFGFLAGS += -DREMAP_DMA_NONCACHED
+CFGFLAGS += -DSAL_SPL_LOCK_ON_IRQ=1
+#CFGFLAGS += -DPTRS_ARE_64BITS
+CFGFLAGS += -DBCM_MONOTONIC_TIME
+
+ifdef DPP_CHIPS
+CFLAGS += -DDUNE_BCM -D__DUNE_LINUX_BCM_CPU_PCIE__ -D__DUNE_LINUX_BCM_CPU_PCP_DMA__
+CFGFLAGS += -DSOC_CM_FUNCTION
+endif
+
+ifdef DFE_CHIPS
+CFLAGS += -DDUNE_BCM -D__DUNE_LINUX_BCM_CPU_PCIE__
+CFGFLAGS += -DSOC_CM_FUNCTION
+endif
+
+ifdef SAND_CHIPS
+#CFLAGS += -D__DUNE_SLK_BCM_CPU__
+endif
+
+# Enable cached DMA memory by default
+ifeq (,$(SAL_BDE_USE_CACHED_DMA_MEM))
+SAL_BDE_USE_CACHED_DMA_MEM = 1
+endif
+ifeq ($(SAL_BDE_USE_CACHED_DMA_MEM),1)
+CFGFLAGS += -DSAL_BDE_CACHE_DMA_MEM
+endif
+
+# Extra variables.
+EXTRA_CFLAGS = -D"KBUILD_STR(s)=\#s" $(basename_flags) $(modname_flags)
+EXTRA_CFLAGS+=-Wno-unused-but-set-variable -Wno-unused-value
+#-Wno-enum-compare -Wno-address -Wno-uninitialized -Wno-sequence-point -Wno-switch
+#EXTRA_CFLAGS+= -mcpu=e500mc -m32 -mhard-float --sysroot=/opt/fsl/1.2/sysroots/ppce500mc-fsl-linux
+
+ARCH = powerpc
+KBUILD_VERBOSE = 1
+
+export ARCH KBUILD_VERBOSE 
+
+
+# From linux/arch/ppc/Makefile
+comma = ,
+basetarget = $(basename $(notdir $@))
+modname = $(basetarget)
+
+name-fix = $(subst $(comma),_,$(subst -,_,$1))
+basename_flags = -D"KBUILD_BASENAME=KBUILD_STR($(call name-fix,$(basetarget)))"
+modname_flags  = $(if $(filter 1,$(words $(modname))),\
+                 -D"KBUILD_MODNAME=KBUILD_STR($(call name-fix,$(modname)))")
+
+
+ifeq (4.0,$(ELDK_VERSION))
+ifeq (ELDK,$(BUILD_PLATFORM))
+KFLAG_INCLD = /tools/eldk/4.0/usr/lib/gcc/powerpc-linux/4.0.0/include
+endif
+endif
+
+ifeq (4.1,$(ELDK_VERSION))
+ifeq (ELDK,$(BUILD_PLATFORM))
+KFLAG_INCLD = /tools/eldk/4.1/usr/lib/gcc/powerpc-linux/4.0.0/include
+endif
+endif
+
+ifeq (2.0,$(WRS_LINUX_VERSION))
+KFLAG_INCLD = /opt/eldk/usr/lib/gcc/powerpc-linux/4.2.2/include
+endif
+
+ifeq (3.0,$(WRS_LINUX_VERSION))
+KFLAG_INCLD = $(WRLINUX_GNU_PATH)/../lib/gcc/powerpc-wrs-linux-gnu/4.3.2/include
+endif
+
+ifdef SBX_CHIPS
+CFLAGS += -DBCM_PLX9656_LOCAL_BUS -DBDE_LINUX_NON_INTERRUPTIBLE 
+endif
+
+ifeq (,$(KFLAGS))
+#JORGE includes acrescentados para o kernel 3.12
+#KFLAGS := -D__KERNEL__ -m32 -nostdinc -isystem $(KFLAG_INCLD) -I$(LINUX_INCLUDE) -include $(LINUX_INCLUDE)/linux/version.h -include $(LINUX_INCLUDE)/generated/autoconf.h -I$(KERNDIR)/arch/powerpc -I$(KERNDIR)/arch/powerpc/include -I$(KERNDIR)/include/asm-generic -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs -O2 -fno-strict-aliasing -fno-common -mhard-float -pipe -ffixed-r2 -mmultiple -mno-altivec -funit-at-a-time -Wa,-me500 -fomit-frame-pointer  -Wdeclaration-after-statement -Wno-pointer-sign
+KFLAGS := -D__KERNEL__ -m32 -nostdinc -isystem $(KFLAG_INCLD) -I$(LINUX_INCLUDE) -I$(LINUX_INCLUDE)/linux -include $(LINUX_INCLUDE)/generated/uapi/linux/version.h -include $(LINUX_INCLUDE)/generated/autoconf.h -I$(LINUX_INCLUDE)/generated/uapi -I$(KERNDIR)/arch/powerpc -I$(KERNDIR)/arch/powerpc/include -I$(KERNDIR)/include/asm-generic -I$(KERNDIR)/arch/powerpc/include/uapi -I$(KERNDIR)/include/uapi -I$(KERNDIR)/arch/powerpc/include/generated -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs -O2 -fno-strict-aliasing -fno-common -mhard-float -pipe -ffixed-r2 -mmultiple -mno-altivec -funit-at-a-time -Wa,-me500 -fomit-frame-pointer  -Wdeclaration-after-statement -Wno-pointer-sign
+endif
+
+# Use MSI interrupts if kernel is compiled with MSI support.
+CFLAGS += -DBDE_LINUX_USE_MSI_INTERRUPT
+
+#Wind river Linux 3.0 needs addtional flags
+ifeq (3.0,$(WRS_LINUX_VERSION))
+KFLAGS += -I$(LINUX_INCLUDE_STD) -I$(KERNDIR_STD)/arch/powerpc -I$(KERNDIR_STD)/arch/powerpc/include -I$(KERNDIR_STD)/include.asm-generic -mno-spe
+endif
+
+ifneq ($(targetplat),user)
+include ${SDK}/make/Makefile.linux-kernel-3_x
+endif
Index: make/Make.local.dnx
===================================================================
--- make/Make.local.dnx	(nonexistent)
+++ make/Make.local.dnx	(revision 3371)
@@ -0,0 +1,598 @@
+# $Id: Make.local.template,v 1.181 Broadcom SDK $
+# $Copyright: Copyright 2012 Broadcom Corporation.
+# This program is the proprietary software of Broadcom Corporation
+# and/or its licensors, and may only be used, duplicated, modified
+# or distributed pursuant to the terms and conditions of a separate,
+# written license agreement executed between you and Broadcom
+# (an "Authorized License").  Except as set forth in an Authorized
+# License, Broadcom grants no license (express or implied), right
+# to use, or waiver of any kind with respect to the Software, and
+# Broadcom expressly reserves all rights in and to the Software
+# and all intellectual property rights therein.  IF YOU HAVE
+# NO AUTHORIZED LICENSE, THEN YOU HAVE NO RIGHT TO USE THIS SOFTWARE
+# IN ANY WAY, AND SHOULD IMMEDIATELY NOTIFY BROADCOM AND DISCONTINUE
+# ALL USE OF THE SOFTWARE.  
+#  
+# Except as expressly set forth in the Authorized License,
+#  
+# 1.     This program, including its structure, sequence and organization,
+# constitutes the valuable trade secrets of Broadcom, and you shall use
+# all reasonable efforts to protect the confidentiality thereof,
+# and to use this information only in connection with your use of
+# Broadcom integrated circuit products.
+#  
+# 2.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, THE SOFTWARE IS
+# PROVIDED "AS IS" AND WITH ALL FAULTS AND BROADCOM MAKES NO PROMISES,
+# REPRESENTATIONS OR WARRANTIES, EITHER EXPRESS, IMPLIED, STATUTORY,
+# OR OTHERWISE, WITH RESPECT TO THE SOFTWARE.  BROADCOM SPECIFICALLY
+# DISCLAIMS ANY AND ALL IMPLIED WARRANTIES OF TITLE, MERCHANTABILITY,
+# NONINFRINGEMENT, FITNESS FOR A PARTICULAR PURPOSE, LACK OF VIRUSES,
+# ACCURACY OR COMPLETENESS, QUIET ENJOYMENT, QUIET POSSESSION OR
+# CORRESPONDENCE TO DESCRIPTION. YOU ASSUME THE ENTIRE RISK ARISING
+# OUT OF USE OR PERFORMANCE OF THE SOFTWARE.
+# 
+# 3.     TO THE MAXIMUM EXTENT PERMITTED BY LAW, IN NO EVENT SHALL
+# BROADCOM OR ITS LICENSORS BE LIABLE FOR (i) CONSEQUENTIAL,
+# INCIDENTAL, SPECIAL, INDIRECT, OR EXEMPLARY DAMAGES WHATSOEVER
+# ARISING OUT OF OR IN ANY WAY RELATING TO YOUR USE OF OR INABILITY
+# TO USE THE SOFTWARE EVEN IF BROADCOM HAS BEEN ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGES; OR (ii) ANY AMOUNT IN EXCESS OF
+# THE AMOUNT ACTUALLY PAID FOR THE SOFTWARE ITSELF OR USD 1.00,
+# WHICHEVER IS GREATER. THESE LIMITATIONS SHALL APPLY NOTWITHSTANDING
+# ANY FAILURE OF ESSENTIAL PURPOSE OF ANY LIMITED REMEDY.$
+#
+
+# Usage for Make.local.template and Make.local:
+#
+# Make.local.template is checked into the tree, but Make.local is never
+# checked in.  Copy Make.local.template to Make.local, then change
+# Make.local to select the desired compilation options (mostly debugging
+# features).
+#
+# Note on CFGFLAGS usage:
+#
+#   Because CFGFLAGS may be used with other source preprocessing tools,
+#   please limit the values added to CFGFLAGS to -D defines.
+#
+#
+# Table of Contents:
+#
+#     Compiler Related Options
+#     Boot and Debug Related Options
+#     Operational and Policy Options
+#     System Management and Multiple CPU Options
+#     Selective Device Support (Switches and PHYs)
+#     Miscellaneous Options
+#
+
+################################################################
+#
+# Compiler Related Options
+#
+################################################################
+
+# Compiling out #ifdef DEBUG code saves about 1.3% on executable size.
+# It is recommended to leave debug enabled when developing applications.
+#DEBUG_IFDEFS=FALSE
+
+# SAL resource usage tracking #ifdef control
+# DEBUG_IFDEFS should also be  TRUE for this switch to take effect.
+#BCM_RESOURCE_USAGE_PROFILE_IFDEFS=TRUE
+
+# Uncomment to build without debug symbols
+#DEBUG_SYMBOLS=FALSE
+
+# Uncomment to add private CFLAGS
+#DEBUG_CFLAGS=
+
+# Uncomment to turn off the optimizer when debugging (recommended)
+#DEBUG_OPTIMIZE=FALSE
+
+# Compiling out assert() saves about 1.1% on executable size,
+# however do so is VERY MUCH discouraged.
+#DEBUG_ASSERTS=FALSE
+
+# Controlling GCC -pedantic flag
+#DEBUG_PEDANTIC=TRUE
+
+# compiler.h overrides, these disable various compiler
+# related features even if the compiler normally supports them
+
+# Disable use of long long for uint64
+#CFGFLAGS += -DCOMPILER_OVERRIDE_NO_LONGLONG
+
+# Disable use of doubles
+#CFGFLAGS += -DCOMPILER_OVERRIDE_NO_DOUBLE
+
+# Disable inlining of functions
+#CFGFLAGS += -DCOMPILER_OVERRIDE_NO_INLINE
+
+# Disable use of const
+#CFGFLAGS += -DCOMPILER_OVERRIDE_NO_CONST
+
+# Disable use of static functions
+#CFGFLAGS += -DCOMPILER_OVERRIDE_NO_STATIC
+
+# Disable RPC flexible (long) bitmaps.
+# Warning: Disabling this changes the RPC version to 2 and
+# makes systems running this code incompatible with systems
+# running previous versions of BCM RPC code.
+#CFGFLAGS += -DBCM_RPC_PBMP_64
+
+#
+# Define if longs are 64 bits on your compiler;
+# this is typically true ONLY for 64-bit processors.
+#CFGFLAGS += -DLONGS_ARE_64BITS
+
+#
+# Define if pointers are 64 bits on your compiler;
+# this is typically true ONLY for 64-bit processors.
+# NOTE: This support works ONLY under conditions where the
+# upper 32 bits for ALL pointers are ZERO.
+#CFGFLAGS += -DPTRS_ARE_64BITS
+
+
+################################################################
+#
+# Boot and Debug Related Options
+#
+################################################################
+
+# Define this to add debug code for RX pool buffer tracking
+#CFGFLAGS += -DBCM_RXP_DEBUG
+
+# Allow debugging of PCI reads/writes (debug +pci)
+#CFGFLAGS += -DSOC_PCI_DEBUG
+
+# Make default debugging settings be 0 (very very quiet)
+#CFGFLAGS += -DNO_DEBUG_OUTPUT_DEFAULT
+
+# Don't use vxMemProbe
+#CFGFLAGS += -DVX_NO_MEM_PROBE
+
+# Allow debugging of Memory Allocation/Deallocation logging
+#CFGFLAGS += -DMEMLOG_SUPPORT
+
+################################################################
+################################################################
+#
+# Operational and Policy Options
+#
+################################################################
+
+# VLAN policy control:
+#	NO_DEFAULT_ETHER	do not init ether ports into vlan 1
+#	NO_DEFAULT_CPU		do not init cpu ports into vlan 1
+#	NO_DEFAULT_SPI_SUBPORT	do not init spi subports into vlan 1
+#	NO_AUTO_STACK		do not init stack/HG ports into created vlans
+#CFGFLAGS += -DBCM_VLAN_NO_DEFAULT_ETHER
+#CFGFLAGS += -DBCM_VLAN_NO_DEFAULT_CPU
+#CFGFLAGS += -DBCM_VLAN_NO_DEFAULT_SPI_SUBPORT
+#CFGFLAGS += -DBCM_VLAN_NO_AUTO_STACK
+
+# VLAN multicast flood Policy Control:
+# BCM_VLAN_MCAST_FLOOD_ALL      Flood all multicast packets to the VLAN
+# BCM_VLAN_MCAST_FLOOD_UNKNOWN  Flood unknown multicast packets to the vlan
+# BCM_VLAN_MCAST_FLOOD_NONE     Forward multicast packets with known
+#                               destination addresses to the appropriate ports.
+#                               All packets destined to an unknown multicast
+#                               address are dropped.
+#CFGFLAGS += -DBCM_MCAST_FLOOD_DEFAULT=BCM_VLAN_MCAST_FLOOD_ALL
+#CFGFLAGS += -DBCM_MCAST_FLOOD_DEFAULT=BCM_VLAN_MCAST_FLOOD_UNKNOWN
+#CFGFLAGS += -DBCM_MCAST_FLOOD_DEFAULT=BCM_VLAN_MCAST_FLOOD_NONE
+
+#Port Enable/Disable Policy control:
+#       PORT_DEFAULT_DISABLE       disable ports during switch initialization
+CFGFLAGS += -DBCM_PORT_DEFAULT_DISABLE
+
+# sal thread priority override (this value used for all threads if defined)
+#CFGFLAGS += -DSAL_THREAD_PRIORITY=255
+
+# disable printing of thread name in messages
+#CFGFLAGS += -DSAL_THREAD_NAME_PRINT_DISABLE
+
+# disable runtime reading of flash config.bcm file (even if !NO_FILEIO)
+CFGFLAGS += -DSAL_CONFIG_FILE=\"/usr/local/var/swdrv-config/config.bcm\" -DSAL_CONFIG_TEMP=\"/usr/local/var/swdrv-config/config.tmp\"
+#-DSAL_CONFIG_FILE_DISABLE
+
+# disable all Application SAL dependencies
+#NO_SAL_APPL=1
+
+
+# Prevent scheduling in SPL locks when interrupt code is run as a thread.
+# This option should not be necessary if all locks are implemented correctly,
+# however, some locks may still rely on the assumption that scheduling does
+# not occur when interrupts are disabled. If you experience any locking
+# problems in e.g. Linux User Mode, try enabling this option.
+# Please note that turning on this option will reduce performance by an
+# estimated 5 to 10 %.
+#CFGFLAGS += -DSAL_SPL_NO_PREEMPT
+
+# disable mapping of higig cosq when mapping priority to cosq
+# (use identity mapping instead : map prio0->cos0, prio1->cos1, ... , prio7->cos7)
+#CFGFLAGS += -DBCM_COSQ_HIGIG_MAP_DISABLE
+
+################################################################
+#
+# System Management and Multiple CPU Options
+#
+################################################################
+
+# Turn on BCMX inclusion
+#     INCLUDE_BCMX for any support
+INCLUDE_BCMX=1
+
+# Telekinesis suite applications for CPU to CPU communication and discovery
+#     CPUDB:         Simple CPU data base manager
+#     CPUTRANS:      CPU to CPU communication mechanisms
+#     DISCOVER:      Simple discovery; will also include CPUTRANS
+#     STKTASK:       Stack manager application
+#
+#CFGFLAGS += -DINCLUDE_LIB_CPUDB
+#CFGFLAGS += -DINCLUDE_LIB_CPUTRANS
+#CFGFLAGS += -DINCLUDE_LIB_DISCOVER
+#CFGFLAGS += -DINCLUDE_LIB_STKTASK
+
+# Option for discovery to use the application-data field in routing
+# packets for Board-ID and CPU base flag information.
+#CFGFLAGS += -DDISCOVER_APP_DATA_BOARDID
+
+# Optionally override list of included dispatch modules
+# Note: including RPC automatically includes the Telekinesis suite libs
+# ESW is the enterprise platforms.
+# ROBO includes the managed devices.
+# SBX includes API support for bcm988020QSK24X2
+#DISPATCH_LIST = RPC ESW ASYNC SBX
+
+# Define to have end-to-end flow control enabled on boards that
+# support it
+#CFGFLAGS += -DBCM_BOARD_AUTO_E2E
+
+# Define the following to support per-CPU transmit pointers.
+# This allows the TX setup and send functions to be defined on a
+# per-CPU basis, allowing mixed in-band and out-of-band communication.
+#CFGFLAGS += -DBCM_C2C_TRANSPORT_SWITCHING
+
+# Options for feature list (INCLUDE_XXX)
+# If FEATURE_LIST is defined, it is a list of features to include.
+# See Make.config for the default feature list.
+#
+# Avaliable features:
+#
+#     BCMX
+#     BCMX_DIAG
+#     CHASSIS
+#     CUSTOMER
+#     DRIVERS
+#     EDITLINE
+#     I2C
+#     L3
+#     MEM_SCAN
+#     ATPTRANS_SOCKET
+#     TELNET
+#     TEST
+#     ACL
+#     RCPU
+#     KNET - Linux user mode kernel network support
+#     BCM_SAL_PROFILE - make available an API to track SAL usage.
+#     CINT - Include the C Interpreter in the diagnostic shell. 
+#            Please note this cannot be used in Linux kernel mode. 
+#     C_UNIT - Include the C unit testing framework.  If CINT is also included
+#              then hooks will be provided for use from it.
+#     PHY_SYM_DBG - PHY GUI MDIO read/write support. Socket interface
+#                   to PHY GUI for Symbolic debugging.
+#     APIMODE - call SDK API functions from shell, requires CINT
+#     DUNE_UI - dune legacy user interface. For debug only. Tested
+#               on linux-user-gto-2.6 only.
+#     KBP   -   include nlm2(11K)/nlm3(12K) KBP support
+#     BHH   -   Include Support for BHH Application (MPLS-TP OAM based on Y.1731) 
+#               using BTE on select devices.
+#     AVS   - Include AVS support
+#FEATURE_LIST=ATPTRANS_SOCKET BCMX_DIAG L3 I2C BCMX MEM_SCAN EDITLINE \
+#    CUSTOMER TELNET DRIVERS CHASSIS TEST ACL RCPU BCM_SAL_PROFILE CINT \
+#    PTP CES FCMAP BOARD KNET REGEX MACSEC APIMODE BFD KBP AVS
+
+###############################################################
+#              KBP supported devices
+#
+#Enable this for 11K device support
+#KBP_DEVICE = KBP_11K
+#
+#Enable this for 12K device support
+#KBP_DEVICE = KBP_ALG
+#
+###############################################################
+
+################################################################
+#
+# Selective Device Support (Switches and PHYs)
+#
+################################################################
+
+# Multiple Chip Support
+#
+# By default, the driver supports all Strata switch and fabric chips
+# included in this software release.  It checks device IDs at runtime
+# to run the correct driver modules.
+#
+# To save space, the driver can be compiled to support just a subset of
+# the chips.  To do this, uncomment the line for BCM_PTL_SPT (partial
+# support) and uncomment one line for each chip to support.
+#
+# Note that there are a lot more chips than drivers.
+# For example, the BCM5615 driver is also used for BCM5625 and BCM5645.
+#
+
+#BCM_PTL_SPT = 1
+
+#BCM_5675_A0 = 1
+#BCM_56102_A0 = 1
+#BCM_56112_A0 = 1
+#BCM_56304_B0 = 1
+#BCM_56314_A0 = 1
+#BCM_56504_A0 = 1
+#BCM_56504_B0 = 1
+#BCM_56514_A0 = 1
+#BCM_56624_A0 = 1
+#BCM_56624_B0 = 1
+#BCM_56680_A0 = 1
+#BCM_56680_B0 = 1
+#BCM_56580_A0 = 1
+#BCM_56700_A0 = 1
+#BCM_56800_A0 = 1
+#BCM_56218_A0 = 1
+#BCM_56224_A0 = 1
+#BCM_56224_B0 = 1
+#BCM_56725_A0 = 1
+#BCM_56820_A0 = 1
+#BCM_53314_A0 = 1
+#BCM_53324_A0 = 1
+#BCM_56634_A0 = 1
+#BCM_56634_B0 = 1
+#BCM_56524_A0 = 1
+#BCM_56524_B0 = 1
+#BCM_56685_A0 = 1
+#BCM_56685_B0 = 1
+#BCM_56334_A0 = 1
+#BCM_56334_B0 = 1
+#BCM_56840_A0 = 1
+#BCM_56840_B0 = 1
+#BCM_56850_A0 = 1
+#BCM_56142_A0 = 1
+#BCM_56150_A0 = 1
+#BCM_56836_A0 = 1
+#BCM_56640_A0 = 1
+#BCM_56440_A0 = 1
+#BCM_56440_B0 = 1
+#BCM_56450_A0 = 1
+#BCM_56960_A0 = 1
+#BCM_56860_A0 = 1
+
+#BCM_5338_A0 = 1
+#BCM_5380_A0 = 1
+#BCM_5338_B0 = 1
+#BCM_5325_A1 = 1
+
+#BCM_5324_A0 = 1
+#BCM_5396_A0 = 1
+#BCM_5389_A0 = 1
+#BCM_5398_A0 = 1
+#BCM_5324_A1 = 1
+#BCM_53115_A0 = 1
+#BCM_53118_A0 = 1
+#BCM_53280_A0 = 1
+#BCM_53280_B0 = 1
+#BCM_53101_A0 = 1
+#BCM_53125_A0 = 1
+#BCM_53128_A0 = 1
+#BCM_53600_A0 = 1
+#BCM_89500_A0 = 1
+
+#BCM_88030_A0 = 1
+#BCM_QE2000_A0 = 1
+#BCM_BME3200_B0 = 1
+#BCM_BM9600_A0 = 1
+#BCM_88230_A0 = 1
+#BCM_88230_B0 = 1
+#BCM_88230_C0 = 1
+
+DPP_CHIPS = 1
+BCM_88375_A0 = 1
+BCM_88375_B0 = 1
+
+BCM_88470_A0 = 1
+BCM_88470_B0 = 1
+BCM_88680_A0 = 1
+BCM_88680_B0 = 1
+
+BCM_88675_A0 = 1
+BCM_88675_B0 = 1
+
+BCM_88650_A0 = 1
+BCM_88650_B0 = 1
+BCM_88660_A0 = 1
+
+BCM_PTL_SPT=1
+DEBUG_OPTIMIZE=FALSE
+#FEATURE_LIST=CINT BSAFE ATPTRANS_SOCKET L3 I2C MEM_SCAN EDITLINE TELNET DRIVERS INTR CHASSIS TEST BCM_SAL_PROFILE RCPU DUNE_UI NO_SAL_APPL=1
+#FEATURE_LIST=ATPTRANS_SOCKET BCMX_DIAG L3 I2C BCMX MEM_SCAN EDITLINE CUSTOMER TEST ACL
+FEATURE_LIST=ATPTRANS_SOCKET BCMX_DIAG L3 I2C BCMX MEM_SCAN EDITLINE CUSTOMER TEST ACL \
+             CINT BCM_SAL_PROFILE NO_SAL_APPL=1
+#TELNET DRIVERS INTR CHASSIS BSAFE RCPU DUNE_UI
+
+#BCM_88640_A0 = 1
+#BCM_88650_A0 = 1
+#BCM_88650_B0 = 1
+#BCM_88650_B1 = 1
+#BCM_88750_A0 = 1
+#BCM_88750_B0 = 1
+
+#BCM_TK371X_A0 = 1
+
+# Options for multiple PHY support
+# If BCM_PHY_LIST is defined, it is a list of PHYs to include.
+# The default is to include all of them.
+# If none of them should be included specify BCM_PHY_LIST=EMPTY
+#BCM_PHY_LIST=522X 54XX 5464 5421S 5482 54616 54680 54680E 52681E 54880E 54682 54684 54640 54640E 54880 SERDES SIMUL 8703 8705 8706 8072 8040 8481 8750 8729 84740 84756 84328 EMPTY
+
+# Options for BCM5338 5380
+#CFGFLAGS += -DROBO_OLD
+#ROBO_OLD = 1
+
+# Support for phy simulation
+#CFGFLAGS += -DINCLUDE_PHY_SIMUL
+#CFGFLAGS += -DSIM_ALL_PHYS        # All phys use simulation driver
+#CFGFLAGS += -DSIM_CMIC_LINK_STAT  # Get link status from CMIC register
+
+# Support for BOARD library
+# if BOARD_LIST is defined, it is a list of Board drivers to include.
+# The default is to include all board drivers appropriate for the devices
+# included in the build. If none of them should be included specify
+# BOARD_LIST=EMPTY.
+#BOARD_LIST=GENERIC
+
+# Support for event logging
+#CFGFLAGS += -DINCLUDE_SHARED_EVLOG
+
+# Support for BCM API port translation
+#CFGFLAGS += -DINCLUDE_BCM_API_XLATE_PORT
+
+# Support for callback error checks and abort in traverse api's
+#CFGFLAGS += -DBCM_CB_ABORT_ON_ERR
+################################################################
+#
+# Misc Options
+#
+################################################################
+#CFGFLAGS += -DSOC_MEM_L3_DEFIP_WAR
+
+# Compile out Register/Table descriptive strings to generate a
+# compact image
+#CFGFLAGS +=-DSOC_NO_NAMES
+#CFGFLAGS +=-DSOC_NO_ALIAS
+#CFGFLAGS +=-DSOC_NO_DESC
+
+# Reload/WarmBoot Support
+#
+#CFGFLAGS += -DBCM_WARM_BOOT_SUPPORT
+#
+# Need this for validation using SOC scripts; Will move to tcl 
+# someday
+#CFGFLAGS += -DBCM_WARM_BOOT_SUPPORT_SW_DUMP
+#
+# Adds a CRC check on scache buffer: Calculate when saving,
+# and verify when loading.
+# When doing ISSU, both source and destination versions should either have
+# this flag enabled or disabled.
+#CFGFLAGS += -DSCACHE_CRC_CHECK
+
+################################################################
+#
+# Enable Easy Reload Support
+#
+################################################################
+#CFGFLAGS += -DBCM_EASY_RELOAD_SUPPORT
+# For validation purposes
+#CFGFLAGS += -DBCM_EASY_RELOAD_SUPPORT_SW_DUMP
+
+# Software Trunk failover Support
+#
+#CFGFLAGS += -DBCM_TRUNK_FAILOVER_SUPPORT
+
+# This option is not present in the default make.local.template.
+ifdef LVL7_FIXUP
+CFGFLAGS += -DBCM_BOARD_AUTO_TRUNK
+endif
+
+################################################################
+#
+# Override default VXWORKS thread options to make set 
+# VX_UNBREAKABLE flag in task creation.
+#
+################################################################
+#CFGFLAGS += -DVX_THREAD_OPT_UNBREAKABLE
+
+
+################################################################
+#
+# Use default priority for BDE interrupt thread.
+#
+################################################################
+#CFGFLAGS += -DSAL_BDE_THREAD_PRIO_DEFAULT
+
+################################################################
+#
+# Use cached DMA memory when mapping kernel DMA memory to user
+# mode. Should only be enabled on cache-coherent platforms.
+#
+################################################################
+#CFGFLAGS += -DSAL_BDE_CACHE_DMA_MEM
+
+################################################################
+#
+# Take the spl lock upon entering an ISR
+#
+################################################################
+#CFGFLAGS += -DSAL_SPL_LOCK_ON_IRQ
+
+################################################################
+#
+# Silently ignore NULL pointer free in sal_free API
+# Default behaviour is to assert if a NULL pointer is passed to sal_free
+#
+################################################################
+#CFGFLAGS += -DSAL_FREE_NULL_IGNORE
+
+################################################################
+# Enable SBX MPLS TP support
+################################################################
+#CFGFLAGS += -DBCM_SBX_MPLSTP_SUPPORT
+#CFGFLAGS += -DBCM_SBX_C1_MPLSTP_SUPPORT
+
+################################################################
+# Restrict SBX C2 Fte range to C2's range
+################################################################
+#CFGFLAGS += -DBCM_SBX_C1_C2_INTEROP
+
+################################################################
+# For historical reasons the PCI probe function skips device 12
+# by default to prevent a system hang on certain platforms.
+# Set this value to zero to probe all PCI devices.
+################################################################
+#CFGFLAGS += -DOVERRIDE_PCI_SKIP_DEV_MASK=0
+
+################################################################
+# Override max devices supported by PLI BDE
+################################################################
+#CFGFLAGS += -DPLI_MAX_DEVICES
+
+################################################################
+# Track BCM API calls to avoid deinitialization while calls active
+# This will incur a small time penalty for each BCM API call
+################################################################
+#CFGFLAGS += -DBCM_CONTROL_API_TRACKING
+
+################################################################
+# Override default retry time for detach to wait for executing
+# APIs to complete.
+################################################################
+#CFGFLAGS += -DBCM_DETACH_POLL_INTERVAL_USECS_DEFAULT=100000
+#CFGFLAGS += -DBCM_DETACH_NUM_RETRIES_DEFAULT=3000
+
+################################################################
+# Disable the RX module initialization
+################################################################
+#CFGFLAGS += -DBCM_RX_DISABLE
+
+################################################################
+# Enable TX callback in interrupt thread
+################################################################
+#CFGFLAGS += -DTX_CB_INTR
+
+
+ifdef LVL7_FIXUP
+# Turn on BCM Shell inclusion
+L7_INCLUDE_BCM_SHELL = 1
+CFGFLAGS += -DL7_BCM_SHELL_SUPPORT
+endif
+
Index: ptin_platforms.txt
===================================================================
--- ptin_platforms.txt	(nonexistent)
+++ ptin_platforms.txt	(revision 3371)
@@ -0,0 +1,10 @@
+./systems/linux/user/cxo2t2-arm64-3_x
+./systems/linux/kernel/cxo2t2-arm64-3_x
+./make/Makefile.linux-cxo2t2-arm64-3_x
+./systems/linux/user/cxo2t4-e500mc-3_x
+./systems/linux/kernel/cxo2t4-e500mc-3_x
+./make/Makefile.linux-cxo2t4-e500mc-3_x
+./make/Makefile.linux-kernel-3_x
+./systems/linux/user/fastpath_common
+./systems/linux/user/fastpath-3_x
+./make/Make.local.dnx
Index: ptin_platforms.tgz
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/octet-stream
Index: ptin_platforms.tgz
===================================================================
--- ptin_platforms.tgz	(nonexistent)
+++ ptin_platforms.tgz	(revision 3371)

Property changes on: ptin_platforms.tgz
___________________________________________________________________
Added: svn:mime-type
## -0,0 +1 ##
+application/octet-stream
\ No newline at end of property
